{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## imports\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKSPACE\n",
      "============================================================\n",
      "Index(['WORKSPACE_ID', 'CATEGORY', 'NAME', 'TYPE'], dtype='object')\n",
      "(163, 4)\n",
      "lk_exp_measurement_tech\n",
      "============================================================\n",
      "Index(['NAME', 'DESCRIPTION', 'LINK'], dtype='object')\n",
      "(92, 3)\n",
      "EXPERIMENT\n",
      "============================================================\n",
      "Index(['EXPERIMENT_ACCESSION', 'DESCRIPTION', 'MEASUREMENT_TECHNIQUE', 'NAME',\n",
      "       'STUDY_ACCESSION', 'WORKSPACE_ID'],\n",
      "      dtype='object')\n",
      "(3047, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAME</th>\n",
       "      <th>DESCRIPTION</th>\n",
       "      <th>LINK</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Fluorescent Antibody Procedure</td>\n",
       "      <td>An immunological procedure in which the antibo...</td>\n",
       "      <td>http://purl.obolibrary.org/obo/NCIT_C17370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Neuraminidase Inhibition Assay</td>\n",
       "      <td>A type of enzyme-based assay that assesses neu...</td>\n",
       "      <td>http://purl.obolibrary.org/obo/NCIT_C120696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Microneutralization Assay</td>\n",
       "      <td>An in vitro assay that is used to determine wh...</td>\n",
       "      <td>http://purl.obolibrary.org/obo/NCIT_C120695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bulk RNA-seq assay</td>\n",
       "      <td>An RNA-seq assay in which the sample that is s...</td>\n",
       "      <td>http://purl.obolibrary.org/obo/OBI_0003090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1D Gel</td>\n",
       "      <td>One dimensional gels are used to separate an a...</td>\n",
       "      <td>http://purl.obolibrary.org/obo/OBI_0001121</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             NAME  \\\n",
       "0  Fluorescent Antibody Procedure   \n",
       "1  Neuraminidase Inhibition Assay   \n",
       "2       Microneutralization Assay   \n",
       "3              Bulk RNA-seq assay   \n",
       "4                          1D Gel   \n",
       "\n",
       "                                         DESCRIPTION  \\\n",
       "0  An immunological procedure in which the antibo...   \n",
       "1  A type of enzyme-based assay that assesses neu...   \n",
       "2  An in vitro assay that is used to determine wh...   \n",
       "3  An RNA-seq assay in which the sample that is s...   \n",
       "4  One dimensional gels are used to separate an a...   \n",
       "\n",
       "                                          LINK  \n",
       "0   http://purl.obolibrary.org/obo/NCIT_C17370  \n",
       "1  http://purl.obolibrary.org/obo/NCIT_C120696  \n",
       "2  http://purl.obolibrary.org/obo/NCIT_C120695  \n",
       "3   http://purl.obolibrary.org/obo/OBI_0003090  \n",
       "4   http://purl.obolibrary.org/obo/OBI_0001121  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "immport_prefix=\"/var/datasets/ImmPort/immune_RNAseq/ALLSTUDIES/ALLSTUDIES-DR49.2_Tab/Tab/\"\n",
    "\n",
    "\n",
    "print(\"WORKSPACE\")\n",
    "print(\"============================================================\")\n",
    "workspace = pd.read_table(f\"{immport_prefix}/workspace.txt\",sep=\"\\t\")\n",
    "print(workspace.columns)\n",
    "print(workspace.shape)\n",
    "\n",
    "\n",
    "### contains elinks\n",
    "print(\"lk_exp_measurement_tech\")\n",
    "print(\"============================================================\")\n",
    "lk_exp_measurement_tech = pd.read_table(f\"{immport_prefix}/lk_exp_measurement_tech.txt\",sep=\"\\t\")\n",
    "print(lk_exp_measurement_tech.columns)\n",
    "print(lk_exp_measurement_tech.shape)\n",
    "\n",
    "print(\"EXPERIMENT\")\n",
    "print(\"============================================================\")\n",
    "experiment = pd.read_table(f\"{immport_prefix}/experiment.txt\",sep=\"\\t\")\n",
    "print(experiment.columns)\n",
    "print(experiment.shape)\n",
    "experiment.head()\n",
    "\n",
    "lk_exp_measurement_tech.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EXPSAMPLE_PLUBLIC_REPOSITORY\n",
      "============================================================\n",
      "Index(['RESULT_ID', 'EXPSAMPLE_ACCESSION', 'REPOSITORY_ACCESSION',\n",
      "       'REPOSITORY_NAME'],\n",
      "      dtype='object')\n",
      "(52404, 4)\n",
      "EXPSAMPLE\n",
      "============================================================\n",
      "Index(['EXPSAMPLE_ACCESSION', 'DESCRIPTION', 'EXPERIMENT_ACCESSION', 'NAME',\n",
      "       'RESULT_SCHEMA', 'UPLOAD_RESULT_STATUS', 'WORKSPACE_ID'],\n",
      "      dtype='object')\n",
      "(850798, 7)\n",
      "EXPERIMENT\n",
      "============================================================\n",
      "Index(['EXPERIMENT_ACCESSION', 'DESCRIPTION', 'MEASUREMENT_TECHNIQUE', 'NAME',\n",
      "       'STUDY_ACCESSION', 'WORKSPACE_ID'],\n",
      "      dtype='object')\n",
      "(3047, 6)\n"
     ]
    }
   ],
   "source": [
    "print(\"EXPSAMPLE_PLUBLIC_REPOSITORY\")\n",
    "print(\"============================================================\")\n",
    "gene_expression = pd.read_table(f\"{immport_prefix}/expsample_public_repository.txt\",sep=\"\\t\")\n",
    "print(gene_expression.columns)\n",
    "print(gene_expression.shape)\n",
    "\n",
    "print(\"EXPSAMPLE\")\n",
    "print(\"============================================================\")\n",
    "expsample = pd.read_table(f\"{immport_prefix}/expsample.txt\",sep=\"\\t\")\n",
    "print(expsample.columns)\n",
    "print(expsample.shape)\n",
    "\n",
    "print(\"EXPERIMENT\")\n",
    "print(\"============================================================\")\n",
    "experiment = pd.read_table(f\"{immport_prefix}/experiment.txt\",sep=\"\\t\")\n",
    "print(experiment.columns)\n",
    "print(experiment.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EXPSAMPLE_ACCESSION</th>\n",
       "      <th>DESCRIPTION</th>\n",
       "      <th>EXPERIMENT_ACCESSION</th>\n",
       "      <th>NAME</th>\n",
       "      <th>RESULT_SCHEMA</th>\n",
       "      <th>UPLOAD_RESULT_STATUS</th>\n",
       "      <th>WORKSPACE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>674909</th>\n",
       "      <td>ES769464</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EXP13591</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>Not_Parsed</td>\n",
       "      <td>2799</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       EXPSAMPLE_ACCESSION DESCRIPTION EXPERIMENT_ACCESSION NAME  \\\n",
       "674909            ES769464         NaN             EXP13591  NaN   \n",
       "\n",
       "       RESULT_SCHEMA UPLOAD_RESULT_STATUS  WORKSPACE_ID  \n",
       "674909         OTHER           Not_Parsed          2799  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expsample[expsample['EXPSAMPLE_ACCESSION']=='ES769464']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EXPERIMENT_ACCESSION</th>\n",
       "      <th>DESCRIPTION</th>\n",
       "      <th>MEASUREMENT_TECHNIQUE</th>\n",
       "      <th>NAME</th>\n",
       "      <th>STUDY_ACCESSION</th>\n",
       "      <th>WORKSPACE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>EXP13591</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Compare whole blood (Tempus tube) to PBMC gene...</td>\n",
       "      <td>SDY690</td>\n",
       "      <td>2799</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    EXPERIMENT_ACCESSION DESCRIPTION             MEASUREMENT_TECHNIQUE  \\\n",
       "262             EXP13591         NaN  Transcription profiling by array   \n",
       "\n",
       "                                                  NAME STUDY_ACCESSION  \\\n",
       "262  Compare whole blood (Tempus tube) to PBMC gene...          SDY690   \n",
       "\n",
       "     WORKSPACE_ID  \n",
       "262          2799  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "experiment[experiment['EXPERIMENT_ACCESSION']=='EXP13591']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(52404, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RESULT_ID</th>\n",
       "      <th>EXPSAMPLE_ACCESSION</th>\n",
       "      <th>REPOSITORY_ACCESSION</th>\n",
       "      <th>REPOSITORY_NAME</th>\n",
       "      <th>DESCRIPTION_x</th>\n",
       "      <th>EXPERIMENT_ACCESSION</th>\n",
       "      <th>NAME_x</th>\n",
       "      <th>RESULT_SCHEMA</th>\n",
       "      <th>UPLOAD_RESULT_STATUS</th>\n",
       "      <th>WORKSPACE_ID_x</th>\n",
       "      <th>DESCRIPTION_y</th>\n",
       "      <th>MEASUREMENT_TECHNIQUE</th>\n",
       "      <th>NAME_y</th>\n",
       "      <th>STUDY_ACCESSION</th>\n",
       "      <th>WORKSPACE_ID_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>ES776590</td>\n",
       "      <td>GSM1422023</td>\n",
       "      <td>GEO</td>\n",
       "      <td>cRNA27760 Positive for globin peak</td>\n",
       "      <td>EXP13635</td>\n",
       "      <td>5920711019_C</td>\n",
       "      <td>PUBLIC REPOSITORY</td>\n",
       "      <td>Parsed</td>\n",
       "      <td>2883</td>\n",
       "      <td>Raw (non-normalized) Illumina data; mapping be...</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Gene expression measurement on 2010 subjects a...</td>\n",
       "      <td>SDY311</td>\n",
       "      <td>2883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ES776602</td>\n",
       "      <td>GSM1422046</td>\n",
       "      <td>GEO</td>\n",
       "      <td>cRNA27783 Positive for globin peak</td>\n",
       "      <td>EXP13635</td>\n",
       "      <td>5920834001_E</td>\n",
       "      <td>PUBLIC REPOSITORY</td>\n",
       "      <td>Parsed</td>\n",
       "      <td>2883</td>\n",
       "      <td>Raw (non-normalized) Illumina data; mapping be...</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Gene expression measurement on 2010 subjects a...</td>\n",
       "      <td>SDY311</td>\n",
       "      <td>2883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>ES776615</td>\n",
       "      <td>GSM1422059</td>\n",
       "      <td>GEO</td>\n",
       "      <td>cRNA27796 Positive for globin peak</td>\n",
       "      <td>EXP13635</td>\n",
       "      <td>5920666013_E</td>\n",
       "      <td>PUBLIC REPOSITORY</td>\n",
       "      <td>Parsed</td>\n",
       "      <td>2883</td>\n",
       "      <td>Raw (non-normalized) Illumina data; mapping be...</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Gene expression measurement on 2010 subjects a...</td>\n",
       "      <td>SDY311</td>\n",
       "      <td>2883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ES776617</td>\n",
       "      <td>GSM1422061</td>\n",
       "      <td>GEO</td>\n",
       "      <td>cRNA27798 Positive for globin peak</td>\n",
       "      <td>EXP13635</td>\n",
       "      <td>5920666014_C</td>\n",
       "      <td>PUBLIC REPOSITORY</td>\n",
       "      <td>Parsed</td>\n",
       "      <td>2883</td>\n",
       "      <td>Raw (non-normalized) Illumina data; mapping be...</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Gene expression measurement on 2010 subjects a...</td>\n",
       "      <td>SDY311</td>\n",
       "      <td>2883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>ES776625</td>\n",
       "      <td>GSM1422069</td>\n",
       "      <td>GEO</td>\n",
       "      <td>cRNA27806 Positive for globin peak</td>\n",
       "      <td>EXP13635</td>\n",
       "      <td>5920666014_K</td>\n",
       "      <td>PUBLIC REPOSITORY</td>\n",
       "      <td>Parsed</td>\n",
       "      <td>2883</td>\n",
       "      <td>Raw (non-normalized) Illumina data; mapping be...</td>\n",
       "      <td>Transcription profiling by array</td>\n",
       "      <td>Gene expression measurement on 2010 subjects a...</td>\n",
       "      <td>SDY311</td>\n",
       "      <td>2883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   RESULT_ID EXPSAMPLE_ACCESSION REPOSITORY_ACCESSION REPOSITORY_NAME  \\\n",
       "0          1            ES776590           GSM1422023             GEO   \n",
       "1          2            ES776602           GSM1422046             GEO   \n",
       "2          3            ES776615           GSM1422059             GEO   \n",
       "3          4            ES776617           GSM1422061             GEO   \n",
       "4          5            ES776625           GSM1422069             GEO   \n",
       "\n",
       "                        DESCRIPTION_x EXPERIMENT_ACCESSION        NAME_x  \\\n",
       "0  cRNA27760 Positive for globin peak             EXP13635  5920711019_C   \n",
       "1  cRNA27783 Positive for globin peak             EXP13635  5920834001_E   \n",
       "2  cRNA27796 Positive for globin peak             EXP13635  5920666013_E   \n",
       "3  cRNA27798 Positive for globin peak             EXP13635  5920666014_C   \n",
       "4  cRNA27806 Positive for globin peak             EXP13635  5920666014_K   \n",
       "\n",
       "       RESULT_SCHEMA UPLOAD_RESULT_STATUS  WORKSPACE_ID_x  \\\n",
       "0  PUBLIC REPOSITORY               Parsed            2883   \n",
       "1  PUBLIC REPOSITORY               Parsed            2883   \n",
       "2  PUBLIC REPOSITORY               Parsed            2883   \n",
       "3  PUBLIC REPOSITORY               Parsed            2883   \n",
       "4  PUBLIC REPOSITORY               Parsed            2883   \n",
       "\n",
       "                                       DESCRIPTION_y  \\\n",
       "0  Raw (non-normalized) Illumina data; mapping be...   \n",
       "1  Raw (non-normalized) Illumina data; mapping be...   \n",
       "2  Raw (non-normalized) Illumina data; mapping be...   \n",
       "3  Raw (non-normalized) Illumina data; mapping be...   \n",
       "4  Raw (non-normalized) Illumina data; mapping be...   \n",
       "\n",
       "              MEASUREMENT_TECHNIQUE  \\\n",
       "0  Transcription profiling by array   \n",
       "1  Transcription profiling by array   \n",
       "2  Transcription profiling by array   \n",
       "3  Transcription profiling by array   \n",
       "4  Transcription profiling by array   \n",
       "\n",
       "                                              NAME_y STUDY_ACCESSION  \\\n",
       "0  Gene expression measurement on 2010 subjects a...          SDY311   \n",
       "1  Gene expression measurement on 2010 subjects a...          SDY311   \n",
       "2  Gene expression measurement on 2010 subjects a...          SDY311   \n",
       "3  Gene expression measurement on 2010 subjects a...          SDY311   \n",
       "4  Gene expression measurement on 2010 subjects a...          SDY311   \n",
       "\n",
       "   WORKSPACE_ID_y  \n",
       "0            2883  \n",
       "1            2883  \n",
       "2            2883  \n",
       "3            2883  \n",
       "4            2883  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expsample_gene_expr = pd.merge(gene_expression, expsample, on='EXPSAMPLE_ACCESSION', how='inner')\n",
    "experiment_sample_gene_expr = pd.merge(expsample_gene_expr, experiment, on='EXPERIMENT_ACCESSION', how='left')\n",
    "print(experiment_sample_gene_expr.shape)\n",
    "experiment_sample_gene_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['SDY311', 'SDY350', 'SDY296', 'SDY305', 'SDY315', 'SDY404',\n",
       "       'SDY400', 'SDY691', 'SDY406', 'SDY113', 'SDY313', 'SDY212',\n",
       "       'SDY111', 'SDY224', 'SDY180', 'SDY63', 'SDY269', 'SDY56', 'SDY460',\n",
       "       'SDY773', 'SDY301', 'SDY508', 'SDY183', 'SDY522', 'SDY144',\n",
       "       'SDY587', 'SDY299', 'SDY622', 'SDY80', 'SDY420', 'SDY720', 'SDY58',\n",
       "       'SDY675', 'SDY789', 'SDY788', 'SDY546', 'SDY1324', 'SDY823',\n",
       "       'SDY984', 'SDY1064', 'SDY1079', 'SDY1085', 'SDY1092', 'SDY1093',\n",
       "       'SDY1099', 'SDY1043', 'SDY951', 'SDY1119', 'SDY1148', 'SDY1155',\n",
       "       'SDY1027', 'SDY1162', 'SDY1157', 'SDY888', 'SDY1185', 'SDY1190',\n",
       "       'SDY997', 'SDY1217', 'SDY903', 'SDY520', 'SDY640', 'SDY998',\n",
       "       'SDY1172', 'SDY1249', 'SDY1298', 'SDY1289', 'SDY1293', 'SDY1276',\n",
       "       'SDY1302', 'SDY1260', 'SDY1294', 'SDY1206', 'SDY1328', 'SDY1264',\n",
       "       'SDY1288', 'SDY1256', 'SDY1364', 'SDY1291', 'SDY1325', 'SDY1370',\n",
       "       'SDY787', 'SDY1373', 'SDY1343', 'SDY270', 'SDY61', 'SDY1368',\n",
       "       'SDY820', 'SDY1230', 'SDY1086', 'SDY1401', 'SDY1402', 'SDY1389',\n",
       "       'SDY616', 'SDY1412', 'SDY1341', 'SDY1267', 'SDY1387', 'SDY1490',\n",
       "       'SDY1439', 'SDY1529', 'SDY1039', 'SDY1536', 'SDY1396', 'SDY1397',\n",
       "       'SDY1381', 'SDY1613', 'SDY1630', 'SDY1654', 'SDY1697', 'SDY1722',\n",
       "       'SDY1723', 'SDY1725', 'SDY1726', 'SDY1600', 'SDY1395', 'SDY1768',\n",
       "       'SDY1393', 'SDY1773', 'SDY1885', 'SDY1912', 'SDY1880', 'SDY1924',\n",
       "       'SDY1928', 'SDY1971', 'SDY1980', 'SDY1991', 'SDY1995', 'SDY1669',\n",
       "       'SDY1903', 'SDY1644', 'SDY2107'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "experiment_sample_gene_expr['STUDY_ACCESSION'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([], dtype=object)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### RNAseq whole blood\n",
    "# experiment_sample_gene_expr[experiment_sample_gene_expr['STUDY_ACCESSION']=='SDY1412']['REPOSITORY_ACCESSION'].unique() ###GPL16791\tIllumina HiSeq 2500 (Homo sapiens)\n",
    "#experiment_sample_gene_expr[experiment_sample_gene_expr['STUDY_ACCESSION']=='SDY1092']['REPOSITORY_ACCESSION'].unique() ###GPL11154\tIllumina HiSeq 2000 (Homo sapiens)\n",
    "#experiment_sample_gene_expr[experiment_sample_gene_expr['STUDY_ACCESSION']=='SDY1172']['REPOSITORY_ACCESSION'].unique() ###GPL11154\tIllumina HiSeq 2000 (Homo sapiens)\n",
    "#experiment_sample_gene_expr[experiment_sample_gene_expr['STUDY_ACCESSION']=='SDY1381']['REPOSITORY_ACCESSION'].unique() ###GPL11154\tIllumina HiSeq 2000 (Homo sapiens)\n",
    "### RNAseq PBMC -> TODO hat keine Repository ID ? wtf... \n",
    "### fokus auf whole blood... \n",
    "#experiment_sample_gene_expr[experiment_sample_gene_expr['STUDY_ACCESSION']=='SDY67']['REPOSITORY_ACCESSION'].unique() ### \n",
    "experiment_sample_gene_expr[experiment_sample_gene_expr['EXPSAMPLE_ACCESSION']=='ES769464']['REPOSITORY_ACCESSION'].unique() ### \n",
    "### nope, but in immport NGS.PTL5892.docx    -> Illumina HiSeq 2000 (Homo sapiens)\n",
    "#### Microarray whole blood: \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "StreamModeError",
     "evalue": "the XML file must be opened in binary mode.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mStreamModeError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[7], line 22\u001b[0m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;66;03m# Example usage:\u001b[39;00m\n\u001b[1;32m     21\u001b[0m accession_numbers \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGSM1234567\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGSM9876543\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m---> 22\u001b[0m \u001b[43mget_platform_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43maccession_numbers\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[7], line 12\u001b[0m, in \u001b[0;36mget_platform_info\u001b[0;34m(accession_numbers)\u001b[0m\n\u001b[1;32m     10\u001b[0m gds_id \u001b[38;5;241m=\u001b[39m record[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIdList\u001b[39m\u001b[38;5;124m\"\u001b[39m][\u001b[38;5;241m0\u001b[39m]\n\u001b[1;32m     11\u001b[0m handle \u001b[38;5;241m=\u001b[39m Entrez\u001b[38;5;241m.\u001b[39mefetch(db\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgds\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28mid\u001b[39m\u001b[38;5;241m=\u001b[39mgds_id, retmode\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mxml\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 12\u001b[0m records \u001b[38;5;241m=\u001b[39m \u001b[43mEntrez\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m record \u001b[38;5;129;01min\u001b[39;00m records:\n\u001b[1;32m     14\u001b[0m     platform \u001b[38;5;241m=\u001b[39m record[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlatform_id\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/Bio/Entrez/__init__.py:518\u001b[0m, in \u001b[0;36mread\u001b[0;34m(source, validate, escape, ignore_errors)\u001b[0m\n\u001b[1;32m    515\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mParser\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m DataHandler\n\u001b[1;32m    517\u001b[0m handler \u001b[38;5;241m=\u001b[39m DataHandler(validate, escape, ignore_errors)\n\u001b[0;32m--> 518\u001b[0m record \u001b[38;5;241m=\u001b[39m \u001b[43mhandler\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    519\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m record\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/Bio/Entrez/Parser.py:396\u001b[0m, in \u001b[0;36mDataHandler.read\u001b[0;34m(self, source)\u001b[0m\n\u001b[1;32m    394\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:  \u001b[38;5;66;03m# not a path, assume we received a stream\u001b[39;00m\n\u001b[1;32m    395\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m source\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;241m0\u001b[39m) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m--> 396\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m StreamModeError(\n\u001b[1;32m    397\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mthe XML file must be opened in binary mode.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    398\u001b[0m         ) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    399\u001b[0m     stream \u001b[38;5;241m=\u001b[39m source\n\u001b[1;32m    400\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m stream\u001b[38;5;241m.\u001b[39mread(\u001b[38;5;241m0\u001b[39m) \u001b[38;5;241m!=\u001b[39m \u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "\u001b[0;31mStreamModeError\u001b[0m: the XML file must be opened in binary mode."
     ]
    }
   ],
   "source": [
    "#### Function that gives me the platform info for my study. \n",
    "from Bio import Entrez\n",
    "\n",
    "def get_platform_info(accession_numbers):\n",
    "    Entrez.email = 'elias.schreiner@uni-tuebingen.de'\n",
    "    for accession in accession_numbers:\n",
    "        handle = Entrez.esearch(db=\"gds\", term=accession)\n",
    "        record = Entrez.read(handle)\n",
    "        if record[\"IdList\"]:\n",
    "            gds_id = record[\"IdList\"][0]\n",
    "            handle = Entrez.efetch(db=\"gds\", id=gds_id, retmode=\"xml\")\n",
    "            records = Entrez.read(handle)\n",
    "            for record in records:\n",
    "                platform = record[\"Platform_id\"]\n",
    "                title = record[\"title\"]\n",
    "                print(f\"Accession: {accession}, Platform: {platform}, Title: {title}\")\n",
    "        else:\n",
    "            print(f\"No information found for accession: {accession}\")\n",
    "\n",
    "# Example usage:\n",
    "accession_numbers = [\"GSM1234567\", \"GSM9876543\"]\n",
    "get_platform_info(accession_numbers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
