{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This is designed to merge the single papers together to assemble one bigger data-csv that we can at some point use for model training"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# General Notes: \n",
    "- To execute the notebook, you frist need to produce the .csv from the underlying papers\n",
    "- I try to convert and select for ENSG gene rows as we got  datasets from Flo with 60k entries which most probably will be me mayjority\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np \n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import sys\n",
    "from scipy.io import mmread\n",
    "from utils.helper import manhattanplot, violinplot_overall, scatter_plot, get_negative_values, METADATA_COLS, reorder_columns_by_metadata_and_gene_counts, convert_gene_names_to_ensembl\n",
    "\n",
    "BASE_PATH = \"/work/transcriptomics_norm_model\""
    "from utils.helper import manhattanplot, violinplot_overall, scatter_plot, get_negative_values, METADATA_COLS, reorder_columns_by_metadata_and_gene_counts, convert_gene_names_to_ensembl\n",
    "\n",
    "BASE_PATH=\"/work/transcriptomics_norm_model\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(119, 60710)\n"
     ]
    }
   ],
   "source": [
    "### Wimmers 2023 (NOT INCLUDED)\n",
    "#NOTE This is 1 of 2 Datasets from this paper. \n",
    "#NOTE Also this datasets should be clarified with Flo first. It´s PBMC but not clear from whom ? (maybe GSE239799 ??? but also different)\n",
    "wimmers_2_df = pd.read_csv(f'{BASE_PATH}/2023_wimmers/2023_Wimmers_combi_df.csv')\n",
    "wimmers_2_df = pd.read_csv(f'{BASE_PATH}/2023_wimmers/2023_Wimmers_combi_df.csv')\n",
    "print(wimmers_2_df.shape)\n",
    "wimmers_2_df.head()\n",
    "PLATFORM_GEO_ID_wimmers_23 = \"GPL24676\"\n",
    "PLATFORM_DESCRIPTION_wimmers_23 = \"Illumina NovaSeq 6000 (Homo sapiens)\"\n",
    "METHOD_wimmers_23 = \"RNAseq\"\n",
    "#TODO in the data itself PBMC is mentioned but https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE239787 it is whole blood \n",
    "TYPE_wimmers_23 = \"blood\"\n",
    "study_accession_wimmers_23 = \"???\"\n",
    "desease_wimmers_23 = \"SARS-CoV-2 / healthy\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 60707)\n",
      "(90, 60694)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>subject_accession</th>\n",
       "      <th>data_accession</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>desease</th>\n",
       "      <th>origin</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>PLATFORM_DESCRIPTION</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273532</th>\n",
       "      <th>ENSG00000276351</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GSE239787</td>\n",
       "      <td>CC0146</td>\n",
       "      <td>GSM7673286</td>\n",
       "      <td>1.900000</td>\n",
       "      <td>Sex: Female</td>\n",
       "      <td>acute-omicron</td>\n",
       "      <td>Contact City: Palo Alto</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>292</td>\n",
       "      <td>110</td>\n",
       "      <td>225</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>GSE239787</td>\n",
       "      <td>CC0261</td>\n",
       "      <td>GSM7673287</td>\n",
       "      <td>1.475000</td>\n",
       "      <td>Sex: Female</td>\n",
       "      <td>acute-omicron</td>\n",
       "      <td>Contact City: Palo Alto</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>179</td>\n",
       "      <td>314</td>\n",
       "      <td>178</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>GSE239787</td>\n",
       "      <td>CC0191</td>\n",
       "      <td>GSM7673288</td>\n",
       "      <td>0.541667</td>\n",
       "      <td>Sex: Male</td>\n",
       "      <td>acute</td>\n",
       "      <td>Contact City: Palo Alto</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>147</td>\n",
       "      <td>414</td>\n",
       "      <td>180</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>GSE239787</td>\n",
       "      <td>CC0022</td>\n",
       "      <td>GSM7673289</td>\n",
       "      <td>1.066667</td>\n",
       "      <td>Sex: Female</td>\n",
       "      <td>acute</td>\n",
       "      <td>Contact City: Palo Alto</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>247</td>\n",
       "      <td>256</td>\n",
       "      <td>86</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>GSE239787</td>\n",
       "      <td>CC0184</td>\n",
       "      <td>GSM7673290</td>\n",
       "      <td>0.525000</td>\n",
       "      <td>Sex: Female</td>\n",
       "      <td>acute</td>\n",
       "      <td>Contact City: Palo Alto</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>196</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60694 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   study_accession subject_accession data_accession       age       gender  \\\n",
       "85       GSE239787            CC0146     GSM7673286  1.900000  Sex: Female   \n",
       "86       GSE239787            CC0261     GSM7673287  1.475000  Sex: Female   \n",
       "87       GSE239787            CC0191     GSM7673288  0.541667    Sex: Male   \n",
       "88       GSE239787            CC0022     GSM7673289  1.066667  Sex: Female   \n",
       "89       GSE239787            CC0184     GSM7673290  0.525000  Sex: Female   \n",
       "\n",
       "          desease                   origin   METHOD         TYPE  \\\n",
       "85  acute-omicron  Contact City: Palo Alto  RNA-Seq  Whole blood   \n",
       "86  acute-omicron  Contact City: Palo Alto  RNA-Seq  Whole blood   \n",
       "87          acute  Contact City: Palo Alto  RNA-Seq  Whole blood   \n",
       "88          acute  Contact City: Palo Alto  RNA-Seq  Whole blood   \n",
       "89          acute  Contact City: Palo Alto  RNA-Seq  Whole blood   \n",
       "\n",
       "     PLATFORM_DESCRIPTION  ... ENSG00000273532  ENSG00000276351  \\\n",
       "85  Illumina NovaSeq 6000  ...               0                0   \n",
       "86  Illumina NovaSeq 6000  ...               0                0   \n",
       "87  Illumina NovaSeq 6000  ...               0                0   \n",
       "88  Illumina NovaSeq 6000  ...               0                0   \n",
       "89  Illumina NovaSeq 6000  ...               0                0   \n",
       "\n",
       "    ENSG00000275661  ENSG00000277856  ENSG00000275063  ENSG00000271254  \\\n",
       "85                0              292              110              225   \n",
       "86                0              179              314              178   \n",
       "87                0              147              414              180   \n",
       "88                0              247              256               86   \n",
       "89                0                0                6              196   \n",
       "\n",
       "    ENSG00000275405  ENSG00000275987  ENSG00000277475  ENSG00000268674  \n",
       "85                3                0                0                0  \n",
       "86                2                0                0                0  \n",
       "87                0                1                5                0  \n",
       "88                1                0                0                0  \n",
       "89                2                0                2                0  \n",
       "\n",
       "[5 rows x 60694 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/work/transcriptomics_norm_model/2023_Wimmers_GSE239787_GEO.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[109], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m### Wimmers 2023 \u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m#NOTE This is 2 of 2 Datasets from this paper. \u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m wimmers_GEO239787_df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43mBASE_PATH\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m/2023_Wimmers_GSE239787_GEO.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex_col\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(wimmers_GEO239787_df\u001b[38;5;241m.\u001b[39mshape)\n\u001b[1;32m      5\u001b[0m wimmers_GEO239787_df \u001b[38;5;241m=\u001b[39m wimmers_GEO239787_df\u001b[38;5;241m.\u001b[39mrename(columns\u001b[38;5;241m=\u001b[39m{\n\u001b[1;32m      6\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mID\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msubject_accession\u001b[39m\u001b[38;5;124m'\u001b[39m, \n\u001b[1;32m      7\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgeo_accession\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mdata_accession\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124minstrument_model\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPLATFORM_DESCRIPTION\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[1;32m     16\u001b[0m })\n",
      "File \u001b[0;32m~/miniconda3/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    936\u001b[0m     dialect,\n\u001b[1;32m    937\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    944\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m    945\u001b[0m )\n\u001b[1;32m    946\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 948\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:611\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    608\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    610\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 611\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    613\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    614\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/miniconda3/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1448\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1445\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1447\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1448\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1705\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1703\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1704\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1705\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1706\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1707\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1708\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1709\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1710\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1711\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1712\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1713\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1714\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1715\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1716\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/miniconda3/envs/merge_env/lib/python3.12/site-packages/pandas/io/common.py:863\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    858\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    859\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    860\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    861\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    862\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 863\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    864\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    865\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    866\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    867\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    868\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    869\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    870\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    871\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    872\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/work/transcriptomics_norm_model/2023_Wimmers_GSE239787_GEO.csv'"
     ]
    }
   ],
   "source": [
    "### Wimmers 2023 \n",
    "#NOTE This is 2 of 2 Datasets from this paper. \n",
    "wimmers_GEO239787_df = pd.read_csv(f'{BASE_PATH}/2023_wimmers/2023_Wimmers_GSE239787_GEO.csv', index_col=0)\n",
    "wimmers_GEO239787_df = pd.read_csv(f'{BASE_PATH}/2023_Wimmers_GSE239787_GEO.csv', index_col=0)\n",
    "print(wimmers_GEO239787_df.shape)\n",
    "wimmers_GEO239787_df = wimmers_GEO239787_df.rename(columns={\n",
    "    'ID':'subject_accession', \n",
    "    'geo_accession':'data_accession',\n",
    "    'characteristics_ch1.4':'age',\n",
    "    'characteristics_ch1.3':'gender',\n",
    "    'Status':'desease', \n",
    "    'contact_city':'origin',\n",
    "    'library_strategy':'METHOD', \n",
    "    'source_name_ch1':'TYPE',\n",
    "    'platform_id':'PLATFORM_GEO_ID',\n",
    "    'instrument_model':'PLATFORM_DESCRIPTION',\n",
    "})\n",
    "wimmers_GEO239787_df['study_accession']='GSE239787'\n",
    "wimmers_GEO239787_df['origin']='Contact City: Palo Alto'\n",
    "wimmers_GEO239787_df['age']=wimmers_GEO239787_df['age'].str.replace('age, months: ', '').astype(float)\n",
    "wimmers_GEO239787_df['age']=wimmers_GEO239787_df['age']/12\n",
    "\n",
    "\n",
    "wimmers_GEO239787_df_reduced = reorder_columns_by_metadata_and_gene_counts(wimmers_GEO239787_df)\n",
    "\n",
    "print(wimmers_GEO239787_df_reduced.shape)\n",
    "wimmers_GEO239787_df_reduced.tail()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_166392/518522322.py:4: DtypeWarning: Columns (60729,60730,60735) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  aranachalam_2020_GSE152418_df = pd.read_csv(f\"{BASE_PATH}/2020_arunachalam_wimmers/2020_aranachalam_GSE152418_combi.csv\", index_col=0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(24, 60739)\n",
      "(24, 60694)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>subject_accession</th>\n",
       "      <th>data_accession</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>desease</th>\n",
       "      <th>origin</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>PLATFORM_DESCRIPTION</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273532</th>\n",
       "      <th>ENSG00000276351</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>75.0</td>\n",
       "      <td>gender: F</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>54.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>75.0</td>\n",
       "      <td>gender: F</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>59.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S152_nCoV006EUHM-Draw-1</td>\n",
       "      <td>GSM4614990</td>\n",
       "      <td>59.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>346</td>\n",
       "      <td>5159</td>\n",
       "      <td>66</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60694 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  study_accession        subject_accession data_accession   age     gender  \\\n",
       "1       GSE152418  S147_nCoV001EUHM-Draw-1     GSM4614986  75.0  gender: F   \n",
       "2       GSE152418  S149_nCoV002EUHM-Draw-2     GSM4614987  54.0  gender: M   \n",
       "3       GSE152418  S150_nCoV003EUHM-Draw-1     GSM4614988  75.0  gender: F   \n",
       "4       GSE152418  S151_nCoV004EUHM-Draw-1     GSM4614989  59.0  gender: M   \n",
       "5       GSE152418  S152_nCoV006EUHM-Draw-1     GSM4614990  59.0  gender: M   \n",
       "\n",
       "                   desease   origin   METHOD  TYPE   PLATFORM_DESCRIPTION  \\\n",
       "1  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "2  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "3  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "4  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "5  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "\n",
       "   ... ENSG00000273532  ENSG00000276351  ENSG00000275661  ENSG00000277856  \\\n",
       "1  ...               0                0                0              174   \n",
       "2  ...               0                0                0              283   \n",
       "3  ...               0                0                0             1365   \n",
       "4  ...               0                0                0              184   \n",
       "5  ...               0                0                0              346   \n",
       "\n",
       "   ENSG00000275063  ENSG00000271254  ENSG00000275405  ENSG00000275987  \\\n",
       "1              151               31               22                0   \n",
       "2             1313               56               13                0   \n",
       "3             5028               42               19                0   \n",
       "4             2240               42                5                0   \n",
       "5             5159               66                4                0   \n",
       "\n",
       "   ENSG00000277475  ENSG00000268674  \n",
       "1                0                0  \n",
       "2                0                0  \n",
       "3                0                0  \n",
       "4                1                0  \n",
       "5                0                0  \n",
       "\n",
       "[5 rows x 60694 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
     "ename": "NameError",
     "evalue": "name 'BASE_PATH' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[101], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# 2020 Arunachalam, Wimmers\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m#NOTE migh had some issues in the prep df file, but the GEO actually did a good job \u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m#NOTE I can probably add 4 more subjects but want to clarify with flo first\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m aranachalam_2020_GSE152418_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_csv(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[43mBASE_PATH\u001b[49m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/2020_arunachalam_wimmers/2020_aranachalam_GSE152418_combi.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m, index_col\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m)\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28mprint\u001b[39m(aranachalam_2020_GSE152418_df\u001b[38;5;241m.\u001b[39mshape)\n\u001b[1;32m      6\u001b[0m aranachalam_2020_GSE152418_df \u001b[38;5;241m=\u001b[39m aranachalam_2020_GSE152418_df\u001b[38;5;241m.\u001b[39mrename(columns\u001b[38;5;241m=\u001b[39m{\n\u001b[1;32m      7\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtitle\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msubject_accession\u001b[39m\u001b[38;5;124m'\u001b[39m, \n\u001b[1;32m      8\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgeo_accession\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mdata_accession\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     16\u001b[0m     \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mplatform_id\u001b[39m\u001b[38;5;124m'\u001b[39m:\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPLATFORM_GEO_ID\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[1;32m     17\u001b[0m })\n",
      "\u001b[0;31mNameError\u001b[0m: name 'BASE_PATH' is not defined"
     ]
    }
   ],
   "source": [
    "# 2020 Arunachalam, Wimmers\n",
    "#NOTE migh had some issues in the prep df file, but the GEO actually did a good job \n",
    "#NOTE I can probably add 4 more subjects but want to clarify with flo first\n",
    "aranachalam_2020_GSE152418_df = pd.read_csv(f\"{BASE_PATH}/2020_arunachalam_wimmers/2020_aranachalam_GSE152418_combi.csv\", index_col=0)\n",
    "aranachalam_2020_GSE152418_df = pd.read_csv(f\"{BASE_PATH}/2020_arunachalam_wimmers/2020_aranachalam_GSE152418_combi.csv\", index_col=0)\n",
    "print(aranachalam_2020_GSE152418_df.shape)\n",
    "aranachalam_2020_GSE152418_df = aranachalam_2020_GSE152418_df.rename(columns={\n",
    "    'title':'subject_accession', \n",
    "    'geo_accession':'data_accession',\n",
    "    'Age':'age',\n",
    "    'characteristics_ch1.2':'gender',\n",
    "    'characteristics_ch1.3':'desease', \n",
    "    'contact_city':'origin',\n",
    "    'library_strategy':'METHOD', \n",
    "    'source_name_ch1':'TYPE',\n",
    "    'instrument_model':'PLATFORM_DESCRIPTION',\n",
    "    'platform_id':'PLATFORM_GEO_ID',\n",
    "})\n",
    "aranachalam_2020_GSE152418_df['study_accession'] = 'GSE152418'\n",
    "aranachalam_2020_GSE152418_df = reorder_columns_by_metadata_and_gene_counts(aranachalam_2020_GSE152418_df)\n",
    "print(aranachalam_2020_GSE152418_df.shape)\n",
    "aranachalam_2020_GSE152418_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(89, 20502)\n",
      "(89, 20256)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>subject_accession</th>\n",
       "      <th>data_accession</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>desease</th>\n",
       "      <th>origin</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>PLATFORM_DESCRIPTION</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000210151</th>\n",
       "      <th>ENSG00000286140</th>\n",
       "      <th>ENSG00000286920</th>\n",
       "      <th>ENSG00000284791</th>\n",
       "      <th>ENSG00000242288</th>\n",
       "      <th>ENSG00000267281</th>\n",
       "      <th>ENSG00000285480</th>\n",
       "      <th>ENSG00000285292</th>\n",
       "      <th>ENSG00000286264</th>\n",
       "      <th>ENSG00000287363</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSE152641</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>113219467</td>\n",
       "      <td>113455421</td>\n",
       "      <td>113523636</td>\n",
       "      <td>113523638</td>\n",
       "      <td>113939925</td>\n",
       "      <td>114108587</td>\n",
       "      <td>114483833</td>\n",
       "      <td>114483834</td>\n",
       "      <td>114841035</td>\n",
       "      <td>114841037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE152641</td>\n",
       "      <td>IMX_sample00001</td>\n",
       "      <td>GSM4622627</td>\n",
       "      <td>age: 46</td>\n",
       "      <td>Sex: Male</td>\n",
       "      <td>disease: Healthy control</td>\n",
       "      <td>Burlingame</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>tissue: Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>1004</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE152641</td>\n",
       "      <td>IMX_sample00002</td>\n",
       "      <td>GSM4622628</td>\n",
       "      <td>age: 61</td>\n",
       "      <td>Sex: Male</td>\n",
       "      <td>disease: Healthy control</td>\n",
       "      <td>Burlingame</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>tissue: Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>1063</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE152641</td>\n",
       "      <td>IMX_sample00003</td>\n",
       "      <td>GSM4622629</td>\n",
       "      <td>age: 37</td>\n",
       "      <td>Sex: Male</td>\n",
       "      <td>disease: Healthy control</td>\n",
       "      <td>Burlingame</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>tissue: Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>993</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE152641</td>\n",
       "      <td>IMX_sample00004</td>\n",
       "      <td>GSM4622630</td>\n",
       "      <td>age: 48</td>\n",
       "      <td>Sex: Male</td>\n",
       "      <td>disease: Healthy control</td>\n",
       "      <td>Burlingame</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>tissue: Whole blood</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>904</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 20256 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  study_accession subject_accession data_accession      age     gender  \\\n",
       "0       GSE152641               NaN            NaN      NaN        NaN   \n",
       "1       GSE152641   IMX_sample00001     GSM4622627  age: 46  Sex: Male   \n",
       "2       GSE152641   IMX_sample00002     GSM4622628  age: 61  Sex: Male   \n",
       "3       GSE152641   IMX_sample00003     GSM4622629  age: 37  Sex: Male   \n",
       "4       GSE152641   IMX_sample00004     GSM4622630  age: 48  Sex: Male   \n",
       "\n",
       "                    desease      origin   METHOD                 TYPE  \\\n",
       "0                       NaN         NaN      NaN                  NaN   \n",
       "1  disease: Healthy control  Burlingame  RNA-Seq  tissue: Whole blood   \n",
       "2  disease: Healthy control  Burlingame  RNA-Seq  tissue: Whole blood   \n",
       "3  disease: Healthy control  Burlingame  RNA-Seq  tissue: Whole blood   \n",
       "4  disease: Healthy control  Burlingame  RNA-Seq  tissue: Whole blood   \n",
       "\n",
       "    PLATFORM_DESCRIPTION  ... ENSG00000210151 ENSG00000286140 ENSG00000286920  \\\n",
       "0                    NaN  ...       113219467       113455421       113523636   \n",
       "1  Illumina NovaSeq 6000  ...               0              12            1004   \n",
       "2  Illumina NovaSeq 6000  ...               0              14            1063   \n",
       "3  Illumina NovaSeq 6000  ...               0              50             993   \n",
       "4  Illumina NovaSeq 6000  ...               0              26             904   \n",
       "\n",
       "  ENSG00000284791 ENSG00000242288 ENSG00000267281 ENSG00000285480  \\\n",
       "0       113523638       113939925       114108587       114483833   \n",
       "1               0               0               5               0   \n",
       "2               0               0               9               1   \n",
       "3               1               0               8               0   \n",
       "4               0               0               0               2   \n",
       "\n",
       "  ENSG00000285292 ENSG00000286264 ENSG00000287363  \n",
       "0       114483834       114841035       114841037  \n",
       "1               0               0               0  \n",
       "2               0               0               0  \n",
       "3               0               0               0  \n",
       "4               0               0               0  \n",
       "\n",
       "[5 rows x 20256 columns]"
      ]
     },
     "execution_count": 113,
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 2021 Thair COVID \n",
    "# NOTE \n",
    "\n",
    "thair_2021_GSE152641_df = pd.read_csv(f\"{BASE_PATH}/2021_thair/thair_combi_covid.csv\", low_memory=False, index_col=0)\n",
    "thair_2021_GSE152641_df = pd.read_csv(f\"{BASE_PATH}/2021_thair/thair_combi_covid.csv\", low_memory=False, index_col=0)\n",
    "print(thair_2021_GSE152641_df.shape)\n",
    "thair_2021_GSE152641_df = thair_2021_GSE152641_df.rename(columns={\n",
    "      'title':'subject_accession', \n",
    "       'geo_accession':'data_accession',\n",
    "       'characteristics_ch1.3':'age',\n",
    "       'characteristics_ch1.4':'gender',\n",
    "       'characteristics_ch1':'desease',\n",
    "      'contact_city':'origin',\n",
    "      'library_strategy':'METHOD',\n",
    "      'characteristics_ch1.2':'TYPE',\n",
    "      'instrument_model':'PLATFORM_DESCRIPTION',\n",
    "      'platform_id':'PLATFORM_GEO_ID',\n",
    "})\n",
    "thair_2021_GSE152641_df['study_accession'] ='GSE152641'\n",
    "thair_2021_GSE152641_ordered_df = reorder_columns_by_metadata_and_gene_counts(thair_2021_GSE152641_df)\n",
    "print(thair_2021_GSE152641_ordered_df.shape)\n",
    "thair_2021_GSE152641_ordered_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(32, 23678)\n",
      "First gene: 38\n",
      "First gene: 11\n",
      "Success rate convertion to ENSAMBLE: 94.67428087986464\n",
      "ENSG00000090686\n",
      "ENSG00000090686\n",
      "(32, 23651)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>subject_accession</th>\n",
       "      <th>data_accession</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>desease</th>\n",
       "      <th>origin</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>PLATFORM_DESCRIPTION</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000275945</th>\n",
       "      <th>ENSG00000275993</th>\n",
       "      <th>ENSG00000276289</th>\n",
       "      <th>ENSG00000278678</th>\n",
       "      <th>CDRT15P9</th>\n",
       "      <th>SOWAHCP2</th>\n",
       "      <th>NCOR1P4</th>\n",
       "      <th>GATD3B</th>\n",
       "      <th>ENSG00000280109</th>\n",
       "      <th>PCSEAT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSE169159</td>\n",
       "      <td>PID2038_BL</td>\n",
       "      <td>GSM5900784</td>\n",
       "      <td>33</td>\n",
       "      <td>Male</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>blood</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GSE169159</td>\n",
       "      <td>PID2043_BL</td>\n",
       "      <td>GSM5900789</td>\n",
       "      <td>25</td>\n",
       "      <td>Male</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>blood</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>118</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GSE169159</td>\n",
       "      <td>PID2000_BL</td>\n",
       "      <td>GSM5900795</td>\n",
       "      <td>30</td>\n",
       "      <td>Male</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>blood</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>139</td>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GSE169159</td>\n",
       "      <td>PID2001_BL</td>\n",
       "      <td>GSM5900801</td>\n",
       "      <td>35</td>\n",
       "      <td>Male</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>blood</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>71</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GSE169159</td>\n",
       "      <td>PID2046_BL</td>\n",
       "      <td>GSM5900807</td>\n",
       "      <td>68</td>\n",
       "      <td>Male</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>blood</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>107</td>\n",
       "      <td>46</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23651 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   study_accession subject_accession data_accession  age gender  desease  \\\n",
       "27       GSE169159        PID2038_BL     GSM5900784   33   Male  Healthy   \n",
       "28       GSE169159        PID2043_BL     GSM5900789   25   Male  Healthy   \n",
       "29       GSE169159        PID2000_BL     GSM5900795   30   Male  Healthy   \n",
       "30       GSE169159        PID2001_BL     GSM5900801   35   Male  Healthy   \n",
       "31       GSE169159        PID2046_BL     GSM5900807   68   Male  Healthy   \n",
       "\n",
       "      origin   METHOD   TYPE PLATFORM_DESCRIPTION  ... ENSG00000275945  \\\n",
       "27  Stanford  RNA-Seq  blood       Illumina MiSeq  ...               0   \n",
       "28  Stanford  RNA-Seq  blood       Illumina MiSeq  ...               0   \n",
       "29  Stanford  RNA-Seq  blood       Illumina MiSeq  ...               2   \n",
       "30  Stanford  RNA-Seq  blood       Illumina MiSeq  ...               1   \n",
       "31  Stanford  RNA-Seq  blood       Illumina MiSeq  ...               7   \n",
       "\n",
       "    ENSG00000275993  ENSG00000276289  ENSG00000278678  CDRT15P9  SOWAHCP2  \\\n",
       "27                5                1                0         0         0   \n",
       "28               13               14                1         0         0   \n",
       "29               11                7                2         0         0   \n",
       "30               10                4                0         0         0   \n",
       "31                6               26                1         0         0   \n",
       "\n",
       "    NCOR1P4  GATD3B  ENSG00000280109  PCSEAT  \n",
       "27        0      31               10       0  \n",
       "28        0     118               25       0  \n",
       "29        0     139               47       2  \n",
       "30        0      71                5       0  \n",
       "31        1     107               46       4  \n",
       "\n",
       "[5 rows x 23651 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 2021 Aranachalam VACCINE\n",
    "#NOTE Had to convert Gene names to ENSEMBL which sometimes results in a not found version -> Flo how to handle \n",
    "#NOTE Vaccine study, i just took the Baseline (blood before the vaccine was given)\n",
    "\n",
    "import copy\n",
    "\n",
    "aranachalam_2021_GSE169159_df = pd.read_csv(f\"{BASE_PATH}/2021_arunachalam/2021_aranachalam_vaccine_combi_df_baseline.csv\")\n",
    "print(aranachalam_2021_GSE169159_df.shape)\n",
    "aranachalam_2021_GSE169159_df = aranachalam_2021_GSE169159_df.rename(columns={\n",
    "        'title':'subject_accession', \n",
    "        'geo_accession':'data_accession',\n",
    "        'age':'age',\n",
    "         'sex':'gender',\n",
    "        'contact_city':'origin',\n",
    "        'library_strategy':'METHOD',\n",
    "        'source_name_ch1':'TYPE',\n",
    "        'instrument_model':'PLATFORM_DESCRIPTION',\n",
    "        'platform_id':'PLATFORM_GEO_ID',\n",
    "})\n",
    "aranachalam_2021_GSE169159_df['desease']='Healthy'\n",
    "aranachalam_2021_GSE169159_df['study_accession']='GSE169159'\n",
    "\n",
    "## This is weird as the overwrite of a subset of columns of a dataframe seems not to work. I tried stuff in different patterns here\n",
    "print(f\"First gene: {aranachalam_2021_GSE169159_df.columns.get_loc('FGR')}\")\n",
    "gene_names_to_convert = aranachalam_2021_GSE169159_df.iloc[:, 38:-2].copy().columns\n",
    "aranachalam_2021_GSE169159_ordered_df = reorder_columns_by_metadata_and_gene_counts(aranachalam_2021_GSE169159_df, \n",
    "                                                                                    metadata_cols=METADATA_COLS,\n",
    "                                                                                    gene_columns=gene_names_to_convert)\n",
    "print(f\"First gene: {aranachalam_2021_GSE169159_ordered_df.columns.get_loc('FGR')}\")\n",
    "converted_gene_names = convert_gene_names_to_ensembl(aranachalam_2021_GSE169159_ordered_df.iloc[:, 11:].copy().columns)\n",
    "\n",
    "# Now i want to overwrite every column of the dataset (also i dont need to use loc or iloc)\n",
    "all_dataset_new_columns = METADATA_COLS + converted_gene_names\n",
    "aranachalam_2021_GSE169159_ordered_df.columns = all_dataset_new_columns\n",
    "\n",
    "print(aranachalam_2021_GSE169159_ordered_df.columns[60])\n",
    "print(converted_gene_names[49]) # as i need to add the meta cols ... \n",
    "\n",
    "print(aranachalam_2021_GSE169159_ordered_df.shape)\n",
    "aranachalam_2021_GSE169159_ordered_df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Concat together"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will contain unprocessed concatinated file with raw counts from rna seq. \n",
    "### TODO´s:\n",
    "- preprocessing\n",
    "- unify entries for everything\n",
    "- NA and NAN in gene counts \n",
    "- douplicates (as there will also be longitudinal data) (check for \"draw as well\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(235, 63131)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>subject_accession</th>\n",
       "      <th>data_accession</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>desease</th>\n",
       "      <th>origin</th>\n",
       "      <th>METHOD</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>PLATFORM_DESCRIPTION</th>\n",
       "      <th>...</th>\n",
       "      <th>H2AZP1</th>\n",
       "      <th>MAPK6P2</th>\n",
       "      <th>SREK1IP1P1</th>\n",
       "      <th>JCADP1</th>\n",
       "      <th>TUBAP1</th>\n",
       "      <th>CDRT15P9</th>\n",
       "      <th>SOWAHCP2</th>\n",
       "      <th>NCOR1P4</th>\n",
       "      <th>GATD3B</th>\n",
       "      <th>PCSEAT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>75.0</td>\n",
       "      <td>gender: F</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>54.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>75.0</td>\n",
       "      <td>gender: F</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>59.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSE152418</td>\n",
       "      <td>S152_nCoV006EUHM-Draw-1</td>\n",
       "      <td>GSM4614990</td>\n",
       "      <td>59.0</td>\n",
       "      <td>gender: M</td>\n",
       "      <td>disease state: COVID-19</td>\n",
       "      <td>Atlanta</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 63131 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  study_accession        subject_accession data_accession   age     gender  \\\n",
       "1       GSE152418  S147_nCoV001EUHM-Draw-1     GSM4614986  75.0  gender: F   \n",
       "2       GSE152418  S149_nCoV002EUHM-Draw-2     GSM4614987  54.0  gender: M   \n",
       "3       GSE152418  S150_nCoV003EUHM-Draw-1     GSM4614988  75.0  gender: F   \n",
       "4       GSE152418  S151_nCoV004EUHM-Draw-1     GSM4614989  59.0  gender: M   \n",
       "5       GSE152418  S152_nCoV006EUHM-Draw-1     GSM4614990  59.0  gender: M   \n",
       "\n",
       "                   desease   origin   METHOD  TYPE   PLATFORM_DESCRIPTION  \\\n",
       "1  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "2  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "3  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "4  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "5  disease state: COVID-19  Atlanta  RNA-Seq  PBMC  Illumina NovaSeq 6000   \n",
       "\n",
       "   ... H2AZP1  MAPK6P2  SREK1IP1P1  JCADP1  TUBAP1  CDRT15P9  SOWAHCP2  \\\n",
       "1  ...    NaN      NaN         NaN     NaN     NaN       NaN       NaN   \n",
       "2  ...    NaN      NaN         NaN     NaN     NaN       NaN       NaN   \n",
       "3  ...    NaN      NaN         NaN     NaN     NaN       NaN       NaN   \n",
       "4  ...    NaN      NaN         NaN     NaN     NaN       NaN       NaN   \n",
       "5  ...    NaN      NaN         NaN     NaN     NaN       NaN       NaN   \n",
       "\n",
       "   NCOR1P4  GATD3B PCSEAT  \n",
       "1      NaN     NaN    NaN  \n",
       "2      NaN     NaN    NaN  \n",
       "3      NaN     NaN    NaN  \n",
       "4      NaN     NaN    NaN  \n",
       "5      NaN     NaN    NaN  \n",
       "\n",
       "[5 rows x 63131 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatinated_df = pd.concat([aranachalam_2020_GSE152418_df, \n",
    "                             wimmers_GEO239787_df_reduced, \n",
    "                             thair_2021_GSE152641_df, \n",
    "                             aranachalam_2021_GSE169159_ordered_df])\n",
    "print(concatinated_df.shape)\n",
    "concatinated_df.head()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "merge_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
