{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GSE169159: blood, healthy afaik, longitudinal \n",
    "GSE171964: Single-cell data, deseased, PBMC\n",
    "\n",
    "Preprocessing bulk: \n",
    "- outlier detection with PCA (3 samples removed as they were more than 1.5sd away)\n",
    "raw counts available \n",
    "- RLE > 0.6 removed \n",
    "- normalised by DESeq2 / BTMsT8\n",
    "\n",
    "Relative log expression (RLE) plots were generated with EDAseq40; samples with an RLE > 0.6 were removed from the analysis. Differential gene analysis was performed using DESeq241 (v.1.26.0), incorporating participant identifier into the model to account for inter-participant bias. Genes were ranked by the Wald statistic as reported by DESeq2 for GSEA using the BTMs18. Per-participant fold changes were computed by dividing the DESeq2 normalized expression data for the day of interest by either day 0 (for day 1, day2 and day 7) or day 21 (for day 22, 28 and 42). To obtain BTM correlates with age, the age of each participant was compared against the per-participant fold changes for day 22. The resulting correlation values were ranked by t-statistic and analysed with GSEA. The same method was employed to obtain BTM correlates with IFNγ. IFN scores were computed by taking the per-participant mean fold change on day 22 of the unique set of genes present in the 5 interferon BTMs (M75, M111.1, M150, M127 and M68) that significantly correlated with day-22 IFNγ fold change. Similarly, the per-participant M16 gene score was computed using average fold change on day 22 of the genes present in M16.\n",
    "\n",
    "\n",
    "Check: GSE97590 -> normalized count matrix from GEO used... -> we see this paper also does not compare microarray data with RNseq\n",
    "(Baseline normalized log2-transformed fold changes were then computed per subject for all genes. GSEA was then performed to identify enriched BTMs using gene lists for each dataset ranked by t-statistic from two-sided Student’s t-tests on the post-vaccination log2-transformed fold changes.)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np \n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "from scipy.io import mmread\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#With his I mostly want to find out how i can integrate this papers data into ImmPort\n",
    "# TODO find out what is _BL (i suspect Baseline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(185, 29)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Age</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>Consent</th>\n",
       "      <th>...</th>\n",
       "      <th>LibrarySource</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>sex</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR18038247</td>\n",
       "      <td>30</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>101</td>\n",
       "      <td>3816732127</td>\n",
       "      <td>PRJNA715457</td>\n",
       "      <td>SAMN25966748</td>\n",
       "      <td>1203925664</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2022-02-25T00:00:00Z</td>\n",
       "      <td>2022-02-15T13:52:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM5900628</td>\n",
       "      <td>female</td>\n",
       "      <td>blood</td>\n",
       "      <td>SRP311216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR18038248</td>\n",
       "      <td>30</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>101</td>\n",
       "      <td>4274587852</td>\n",
       "      <td>PRJNA715457</td>\n",
       "      <td>SAMN25966749</td>\n",
       "      <td>1351097515</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2022-02-25T00:00:00Z</td>\n",
       "      <td>2022-02-15T13:50:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM5900629</td>\n",
       "      <td>female</td>\n",
       "      <td>blood</td>\n",
       "      <td>SRP311216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR18038249</td>\n",
       "      <td>30</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>101</td>\n",
       "      <td>4188481110</td>\n",
       "      <td>PRJNA715457</td>\n",
       "      <td>SAMN25966750</td>\n",
       "      <td>1314161957</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2022-02-25T00:00:00Z</td>\n",
       "      <td>2022-02-15T13:49:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM5900630</td>\n",
       "      <td>female</td>\n",
       "      <td>blood</td>\n",
       "      <td>SRP311216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR18038250</td>\n",
       "      <td>30</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>101</td>\n",
       "      <td>4373580376</td>\n",
       "      <td>PRJNA715457</td>\n",
       "      <td>SAMN25966751</td>\n",
       "      <td>1370039906</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2022-02-25T00:00:00Z</td>\n",
       "      <td>2022-02-15T13:49:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM5900631</td>\n",
       "      <td>female</td>\n",
       "      <td>blood</td>\n",
       "      <td>SRP311216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR18038251</td>\n",
       "      <td>30</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>101</td>\n",
       "      <td>3491091866</td>\n",
       "      <td>PRJNA715457</td>\n",
       "      <td>SAMN25966752</td>\n",
       "      <td>1095712135</td>\n",
       "      <td>GEO</td>\n",
       "      <td>public</td>\n",
       "      <td>...</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2022-02-25T00:00:00Z</td>\n",
       "      <td>2022-02-15T13:46:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM5900632</td>\n",
       "      <td>female</td>\n",
       "      <td>blood</td>\n",
       "      <td>SRP311216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Run  Age Assay Type  AvgSpotLen       Bases   BioProject  \\\n",
       "0  SRR18038247   30    RNA-Seq         101  3816732127  PRJNA715457   \n",
       "1  SRR18038248   30    RNA-Seq         101  4274587852  PRJNA715457   \n",
       "2  SRR18038249   30    RNA-Seq         101  4188481110  PRJNA715457   \n",
       "3  SRR18038250   30    RNA-Seq         101  4373580376  PRJNA715457   \n",
       "4  SRR18038251   30    RNA-Seq         101  3491091866  PRJNA715457   \n",
       "\n",
       "      BioSample       Bytes Center Name Consent  ...   LibrarySource  \\\n",
       "0  SAMN25966748  1203925664         GEO  public  ...  TRANSCRIPTOMIC   \n",
       "1  SAMN25966749  1351097515         GEO  public  ...  TRANSCRIPTOMIC   \n",
       "2  SAMN25966750  1314161957         GEO  public  ...  TRANSCRIPTOMIC   \n",
       "3  SAMN25966751  1370039906         GEO  public  ...  TRANSCRIPTOMIC   \n",
       "4  SAMN25966752  1095712135         GEO  public  ...  TRANSCRIPTOMIC   \n",
       "\n",
       "       Organism  Platform           ReleaseDate           create_date version  \\\n",
       "0  Homo sapiens  ILLUMINA  2022-02-25T00:00:00Z  2022-02-15T13:52:00Z       1   \n",
       "1  Homo sapiens  ILLUMINA  2022-02-25T00:00:00Z  2022-02-15T13:50:00Z       1   \n",
       "2  Homo sapiens  ILLUMINA  2022-02-25T00:00:00Z  2022-02-15T13:49:00Z       1   \n",
       "3  Homo sapiens  ILLUMINA  2022-02-25T00:00:00Z  2022-02-15T13:49:00Z       1   \n",
       "4  Homo sapiens  ILLUMINA  2022-02-25T00:00:00Z  2022-02-15T13:46:00Z       1   \n",
       "\n",
       "  Sample Name     sex source_name  SRA Study  \n",
       "0  GSM5900628  female       blood  SRP311216  \n",
       "1  GSM5900629  female       blood  SRP311216  \n",
       "2  GSM5900630  female       blood  SRP311216  \n",
       "3  GSM5900631  female       blood  SRP311216  \n",
       "4  GSM5900632  female       blood  SRP311216  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Testwise integration of 2020_Arachandra_Wimmers\n",
    "root_folder = '/var/datasets/single_papers/2021_Arunachalam_vaccine/'\n",
    "\n",
    "metadata_df = pd.read_csv(f'{root_folder}/metadata.txt')\n",
    "print(metadata_df.shape)\n",
    "metadata_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(23640, 185)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PID2053_BL</th>\n",
       "      <th>PID2053_Day1.2</th>\n",
       "      <th>PID2053_Day7</th>\n",
       "      <th>PID2053_Day21</th>\n",
       "      <th>PID2053_Day22.23</th>\n",
       "      <th>PID2053_Day28</th>\n",
       "      <th>PID2028_BL</th>\n",
       "      <th>PID2028_Day1.2</th>\n",
       "      <th>PID2028_Day7</th>\n",
       "      <th>PID2028_Day21</th>\n",
       "      <th>...</th>\n",
       "      <th>PID2001_Day7</th>\n",
       "      <th>PID2001_Day21</th>\n",
       "      <th>PID2001_Day22.23</th>\n",
       "      <th>PID2001_Day28</th>\n",
       "      <th>PID2046_BL</th>\n",
       "      <th>PID2046_Day1.2</th>\n",
       "      <th>PID2046_Day7</th>\n",
       "      <th>PID2046_Day21</th>\n",
       "      <th>PID2046_Day22.23</th>\n",
       "      <th>PID2046_Day28</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>FGR</th>\n",
       "      <td>16312</td>\n",
       "      <td>20268</td>\n",
       "      <td>13433</td>\n",
       "      <td>15317</td>\n",
       "      <td>18685</td>\n",
       "      <td>19379</td>\n",
       "      <td>11275</td>\n",
       "      <td>22251</td>\n",
       "      <td>939</td>\n",
       "      <td>11231</td>\n",
       "      <td>...</td>\n",
       "      <td>33607</td>\n",
       "      <td>21610</td>\n",
       "      <td>37503</td>\n",
       "      <td>39622</td>\n",
       "      <td>30299</td>\n",
       "      <td>19013</td>\n",
       "      <td>56015</td>\n",
       "      <td>16061</td>\n",
       "      <td>23595</td>\n",
       "      <td>23312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STPG1</th>\n",
       "      <td>137</td>\n",
       "      <td>150</td>\n",
       "      <td>154</td>\n",
       "      <td>113</td>\n",
       "      <td>136</td>\n",
       "      <td>179</td>\n",
       "      <td>92</td>\n",
       "      <td>108</td>\n",
       "      <td>105</td>\n",
       "      <td>126</td>\n",
       "      <td>...</td>\n",
       "      <td>86</td>\n",
       "      <td>55</td>\n",
       "      <td>62</td>\n",
       "      <td>63</td>\n",
       "      <td>92</td>\n",
       "      <td>111</td>\n",
       "      <td>246</td>\n",
       "      <td>60</td>\n",
       "      <td>117</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NIPAL3</th>\n",
       "      <td>2247</td>\n",
       "      <td>2084</td>\n",
       "      <td>1808</td>\n",
       "      <td>1692</td>\n",
       "      <td>1576</td>\n",
       "      <td>2349</td>\n",
       "      <td>1223</td>\n",
       "      <td>1245</td>\n",
       "      <td>735</td>\n",
       "      <td>1197</td>\n",
       "      <td>...</td>\n",
       "      <td>1404</td>\n",
       "      <td>844</td>\n",
       "      <td>1023</td>\n",
       "      <td>782</td>\n",
       "      <td>847</td>\n",
       "      <td>1162</td>\n",
       "      <td>1903</td>\n",
       "      <td>781</td>\n",
       "      <td>1241</td>\n",
       "      <td>1239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KDM1A</th>\n",
       "      <td>1148</td>\n",
       "      <td>1047</td>\n",
       "      <td>876</td>\n",
       "      <td>813</td>\n",
       "      <td>1049</td>\n",
       "      <td>1133</td>\n",
       "      <td>773</td>\n",
       "      <td>778</td>\n",
       "      <td>167</td>\n",
       "      <td>620</td>\n",
       "      <td>...</td>\n",
       "      <td>1270</td>\n",
       "      <td>676</td>\n",
       "      <td>605</td>\n",
       "      <td>695</td>\n",
       "      <td>1019</td>\n",
       "      <td>988</td>\n",
       "      <td>1748</td>\n",
       "      <td>698</td>\n",
       "      <td>1306</td>\n",
       "      <td>930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E2F2</th>\n",
       "      <td>739</td>\n",
       "      <td>705</td>\n",
       "      <td>1312</td>\n",
       "      <td>1290</td>\n",
       "      <td>1878</td>\n",
       "      <td>1111</td>\n",
       "      <td>1189</td>\n",
       "      <td>1254</td>\n",
       "      <td>282</td>\n",
       "      <td>763</td>\n",
       "      <td>...</td>\n",
       "      <td>1302</td>\n",
       "      <td>719</td>\n",
       "      <td>255</td>\n",
       "      <td>677</td>\n",
       "      <td>1150</td>\n",
       "      <td>1197</td>\n",
       "      <td>3270</td>\n",
       "      <td>1622</td>\n",
       "      <td>1226</td>\n",
       "      <td>2080</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 185 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PID2053_BL  PID2053_Day1.2  PID2053_Day7  PID2053_Day21  \\\n",
       "gene                                                              \n",
       "FGR          16312           20268         13433          15317   \n",
       "STPG1          137             150           154            113   \n",
       "NIPAL3        2247            2084          1808           1692   \n",
       "KDM1A         1148            1047           876            813   \n",
       "E2F2           739             705          1312           1290   \n",
       "\n",
       "        PID2053_Day22.23  PID2053_Day28  PID2028_BL  PID2028_Day1.2  \\\n",
       "gene                                                                  \n",
       "FGR                18685          19379       11275           22251   \n",
       "STPG1                136            179          92             108   \n",
       "NIPAL3              1576           2349        1223            1245   \n",
       "KDM1A               1049           1133         773             778   \n",
       "E2F2                1878           1111        1189            1254   \n",
       "\n",
       "        PID2028_Day7  PID2028_Day21  ...  PID2001_Day7  PID2001_Day21  \\\n",
       "gene                                 ...                                \n",
       "FGR              939          11231  ...         33607          21610   \n",
       "STPG1            105            126  ...            86             55   \n",
       "NIPAL3           735           1197  ...          1404            844   \n",
       "KDM1A            167            620  ...          1270            676   \n",
       "E2F2             282            763  ...          1302            719   \n",
       "\n",
       "        PID2001_Day22.23  PID2001_Day28  PID2046_BL  PID2046_Day1.2  \\\n",
       "gene                                                                  \n",
       "FGR                37503          39622       30299           19013   \n",
       "STPG1                 62             63          92             111   \n",
       "NIPAL3              1023            782         847            1162   \n",
       "KDM1A                605            695        1019             988   \n",
       "E2F2                 255            677        1150            1197   \n",
       "\n",
       "        PID2046_Day7  PID2046_Day21  PID2046_Day22.23  PID2046_Day28  \n",
       "gene                                                                  \n",
       "FGR            56015          16061             23595          23312  \n",
       "STPG1            246             60               117             97  \n",
       "NIPAL3          1903            781              1241           1239  \n",
       "KDM1A           1748            698              1306            930  \n",
       "E2F2            3270           1622              1226           2080  \n",
       "\n",
       "[5 rows x 185 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_df = pd.read_csv(f'{root_folder}/GSE169159_raw_counts_with_genenames.csv', index_col='gene')\n",
    "print(counts_df.shape)\n",
    "counts_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Problem: SRA Toolkit metadata does not have the study Subject accessions ...  Let´s see how else i can get them \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
