{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Michlmayr\n",
    "Only virus data! \n",
    "\n",
    "## Data processing\t\n",
    "Sequencer-generated base call (BCL) files were converted to FASTQ files and the multiplexed samples were separated using bcl2fastq\n",
    "Demultiplexed fastq were assessed for sequencing quality using FastQC (version 0.11.4, http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)\n",
    "The fastq files were quality filtered by using FASTAX Toolkit (http://hannonlab.cshl.edu/fastx_toolkit/) with the invocation fastq_quality_filter -q 30 -p 50 -v -Q 33\n",
    "Sequencing reads that met all quality control requirements were aligned to the latest human reference genome (GRCh38) using HISAT2 (version 2.0.4)\n",
    "\n",
    "Gene expression in units of overlapping read counts per million were obtained by using the htseq-count script from HTseq on SAM files of preprocessed RNA-seq alignments.\n",
    "Counts per million were normalized using the voom methodology and expression profiles were modeled using mixed-effect models in limma (version 3.30.6)\n",
    "Genome_build: GRCh38\n",
    "\n",
    "\n",
    "Supplementary_files_format_and_content: csv file with the following output: lgFCH_AcuteVsConv: log2 of the fold change in expression between acute and convalescent timepoints, FCH_AcuteVsConv: fold change in expression between acute and convalescent timepoints, pvals_AcuteVsConv, fdrs_AcuteVsConv, StatusFCH2FDR0.05_AcuteVsConv, a cutoff for |fold change| >2 and FDR <0.05. It is -1 if the gene is fold change <-2 and FDR <0.05, and it is +1 if the gene is fold change >2 and FDR <0.05. Otherwise it is 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np \n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import sys\n",
    "from scipy.io import mmread\n",
    "sys.path.append(os.path.abspath(os.path.join('..', 'utils')))\n",
    "import geo_utils\n",
    "from geo_utils import read_prefixes_from_metadata_txt\n",
    "\n",
    "root_path = '/var/datasets/single_papers/2018_michlmayr'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(172, 46)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/eliasschreiner/PROJEKT/utils/geo_utils.py:46: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  df = df.applymap(lambda x: x.strip('\"') if isinstance(x, str) else x)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>contact_zip/postal_code</th>\n",
       "      <th>contact_country</th>\n",
       "      <th>data_row_count</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>relation</th>\n",
       "      <th>relation.2</th>\n",
       "      <th>supplementary_file_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Patient 1, acute sample, technical replicate 1</td>\n",
       "      <td>GSM2667266</td>\n",
       "      <td>Public on Jun 13 2018</td>\n",
       "      <td>Jun 13 2017</td>\n",
       "      <td>May 15 2019</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>timepoint: 1-2d post symptom onset</td>\n",
       "      <td>...</td>\n",
       "      <td>10029</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Patient 1, acute sample, technical replicate 2</td>\n",
       "      <td>GSM2667267</td>\n",
       "      <td>Public on Jun 13 2018</td>\n",
       "      <td>Jun 13 2017</td>\n",
       "      <td>May 15 2019</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>timepoint: 1-2d post symptom onset</td>\n",
       "      <td>...</td>\n",
       "      <td>10029</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Patient 1, conv sample, technical replicate 1</td>\n",
       "      <td>GSM2667268</td>\n",
       "      <td>Public on Jun 13 2018</td>\n",
       "      <td>Jun 13 2017</td>\n",
       "      <td>May 15 2019</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>timepoint: 15-17d post symptom onset</td>\n",
       "      <td>...</td>\n",
       "      <td>10029</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Patient 1, conv sample, technical replicate 2</td>\n",
       "      <td>GSM2667269</td>\n",
       "      <td>Public on Jun 13 2018</td>\n",
       "      <td>Jun 13 2017</td>\n",
       "      <td>May 15 2019</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>timepoint: 15-17d post symptom onset</td>\n",
       "      <td>...</td>\n",
       "      <td>10029</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Patient 2, acute sample, technical replicate 1</td>\n",
       "      <td>GSM2667270</td>\n",
       "      <td>Public on Jun 13 2018</td>\n",
       "      <td>Jun 13 2017</td>\n",
       "      <td>May 15 2019</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Whole blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>timepoint: 1-2d post symptom onset</td>\n",
       "      <td>...</td>\n",
       "      <td>10029</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina HiSeq 4000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            title geo_accession  \\\n",
       "0  Patient 1, acute sample, technical replicate 1    GSM2667266   \n",
       "1  Patient 1, acute sample, technical replicate 2    GSM2667267   \n",
       "2   Patient 1, conv sample, technical replicate 1    GSM2667268   \n",
       "3   Patient 1, conv sample, technical replicate 2    GSM2667269   \n",
       "4  Patient 2, acute sample, technical replicate 1    GSM2667270   \n",
       "\n",
       "                  status submission_date last_update_date type channel_count  \\\n",
       "0  Public on Jun 13 2018     Jun 13 2017      May 15 2019  SRA             1   \n",
       "1  Public on Jun 13 2018     Jun 13 2017      May 15 2019  SRA             1   \n",
       "2  Public on Jun 13 2018     Jun 13 2017      May 15 2019  SRA             1   \n",
       "3  Public on Jun 13 2018     Jun 13 2017      May 15 2019  SRA             1   \n",
       "4  Public on Jun 13 2018     Jun 13 2017      May 15 2019  SRA             1   \n",
       "\n",
       "  source_name_ch1  organism_ch1                   characteristics_ch1  ...  \\\n",
       "0     Whole blood  Homo sapiens    timepoint: 1-2d post symptom onset  ...   \n",
       "1     Whole blood  Homo sapiens    timepoint: 1-2d post symptom onset  ...   \n",
       "2     Whole blood  Homo sapiens  timepoint: 15-17d post symptom onset  ...   \n",
       "3     Whole blood  Homo sapiens  timepoint: 15-17d post symptom onset  ...   \n",
       "4     Whole blood  Homo sapiens    timepoint: 1-2d post symptom onset  ...   \n",
       "\n",
       "  contact_zip/postal_code contact_country data_row_count     instrument_model  \\\n",
       "0                   10029             USA              0  Illumina HiSeq 4000   \n",
       "1                   10029             USA              0  Illumina HiSeq 4000   \n",
       "2                   10029             USA              0  Illumina HiSeq 4000   \n",
       "3                   10029             USA              0  Illumina HiSeq 4000   \n",
       "4                   10029             USA              0  Illumina HiSeq 4000   \n",
       "\n",
       "  library_selection  library_source library_strategy  \\\n",
       "0              cDNA  transcriptomic          RNA-Seq   \n",
       "1              cDNA  transcriptomic          RNA-Seq   \n",
       "2              cDNA  transcriptomic          RNA-Seq   \n",
       "3              cDNA  transcriptomic          RNA-Seq   \n",
       "4              cDNA  transcriptomic          RNA-Seq   \n",
       "\n",
       "                                            relation  \\\n",
       "0  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "1  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "2  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "3  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "4  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "\n",
       "                                          relation.2 supplementary_file_1  \n",
       "0  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "1  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "2  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "3  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "4  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "\n",
       "[5 rows x 46 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = read_prefixes_from_metadata_txt(f'{root_path}/GSE99992_series_matrix.txt')\n",
    "print(metadata.shape)\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(53971, 10)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>target_id</th>\n",
       "      <th>ens_gene</th>\n",
       "      <th>ext_gene</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>lgFCH_AcuteVsConv</th>\n",
       "      <th>FCH_AcuteVsConv</th>\n",
       "      <th>pvals_AcuteVsConv</th>\n",
       "      <th>fdrs_AcuteVsConv</th>\n",
       "      <th>StatusFCH2FDR0.05_AcuteVsConv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000000412</td>\n",
       "      <td>ENST00000000412</td>\n",
       "      <td>ENSG00000003056</td>\n",
       "      <td>M6PR</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-0.11</td>\n",
       "      <td>-1.08</td>\n",
       "      <td>4.990000e-01</td>\n",
       "      <td>5.750000e-01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000000442</td>\n",
       "      <td>ENST00000000442</td>\n",
       "      <td>ENSG00000173153</td>\n",
       "      <td>ESRRA</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1.09</td>\n",
       "      <td>5.250000e-01</td>\n",
       "      <td>5.990000e-01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000001008</td>\n",
       "      <td>ENST00000001008</td>\n",
       "      <td>ENSG00000004478</td>\n",
       "      <td>FKBP4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.39</td>\n",
       "      <td>1.31</td>\n",
       "      <td>2.510000e-04</td>\n",
       "      <td>6.140000e-04</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000002125</td>\n",
       "      <td>ENST00000002125</td>\n",
       "      <td>ENSG00000003509</td>\n",
       "      <td>NDUFAF7</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-0.04</td>\n",
       "      <td>-1.03</td>\n",
       "      <td>8.960000e-01</td>\n",
       "      <td>9.190000e-01</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000002165</td>\n",
       "      <td>ENST00000002165</td>\n",
       "      <td>ENSG00000001036</td>\n",
       "      <td>FUCA2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>0.42</td>\n",
       "      <td>1.33</td>\n",
       "      <td>1.890000e-25</td>\n",
       "      <td>3.290000e-24</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0        target_id         ens_gene ext_gene    gene_biotype  \\\n",
       "0  ENST00000000412  ENST00000000412  ENSG00000003056     M6PR  protein_coding   \n",
       "1  ENST00000000442  ENST00000000442  ENSG00000173153    ESRRA  protein_coding   \n",
       "2  ENST00000001008  ENST00000001008  ENSG00000004478    FKBP4  protein_coding   \n",
       "3  ENST00000002125  ENST00000002125  ENSG00000003509  NDUFAF7  protein_coding   \n",
       "4  ENST00000002165  ENST00000002165  ENSG00000001036    FUCA2  protein_coding   \n",
       "\n",
       "   lgFCH_AcuteVsConv  FCH_AcuteVsConv  pvals_AcuteVsConv  fdrs_AcuteVsConv  \\\n",
       "0              -0.11            -1.08       4.990000e-01      5.750000e-01   \n",
       "1               0.12             1.09       5.250000e-01      5.990000e-01   \n",
       "2               0.39             1.31       2.510000e-04      6.140000e-04   \n",
       "3              -0.04            -1.03       8.960000e-01      9.190000e-01   \n",
       "4               0.42             1.33       1.890000e-25      3.290000e-24   \n",
       "\n",
       "   StatusFCH2FDR0.05_AcuteVsConv  \n",
       "0                              0  \n",
       "1                              0  \n",
       "2                              0  \n",
       "3                              0  \n",
       "4                              0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_counts = pd.read_csv(f'{root_path}/GSE99992_BigTab.Timepoint.AllGenes.2017-04-25_Transcripts.csv')\n",
    "print(raw_counts.shape)\n",
    "raw_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "merge_env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
