{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One dataset fits well but lacka age information (GSE152418)\n",
    "        Werde  nicht alles mappen können, habe zwar eine xlsx file von flo bekommen, dabei sind aber nicht alle samples enthalten (nCov034) z.b. nicht... denke ich kann ein mapping hinbekommen, aber mit abstrichen\n",
    "The other one is single-cell but does not lack ages (fokus for now is bulk)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np \n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import sys\n",
    "from scipy.io import mmread\n",
    "sys.path.append(os.path.abspath(os.path.join('..', 'utils')))\n",
    "import geo_utils\n",
    "from geo_utils import read_prefixes_from_metadata_txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Testwise integration of 2020_Arachandra_Wimmers\n",
    "root_folder = '/var/datasets/single_papers/2020_Arunachalam_Wimmers/'\n",
    "gse152418_folder = root_folder + 'GSE152418/'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60683, 35)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ENSEMBLID</th>\n",
       "      <th>ENSG00000223972</th>\n",
       "      <th>ENSG00000227232</th>\n",
       "      <th>ENSG00000278267</th>\n",
       "      <th>ENSG00000243485</th>\n",
       "      <th>ENSG00000284332</th>\n",
       "      <th>ENSG00000237613</th>\n",
       "      <th>ENSG00000268020</th>\n",
       "      <th>ENSG00000240361</th>\n",
       "      <th>ENSG00000186092</th>\n",
       "      <th>ENSG00000238009</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273532</th>\n",
       "      <th>ENSG00000276351</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S145_nCOV001_C</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S147_nCoV001EUHM-Draw-1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S149_nCoV002EUHM-Draw-2</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S150_nCoV003EUHM-Draw-1</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S151_nCoV004EUHM-Draw-1</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60683 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ENSEMBLID               ENSG00000223972 ENSG00000227232 ENSG00000278267  \\\n",
       "S145_nCOV001_C                        0               1               3   \n",
       "S147_nCoV001EUHM-Draw-1               0               0               1   \n",
       "S149_nCoV002EUHM-Draw-2               0               3               2   \n",
       "S150_nCoV003EUHM-Draw-1               0              16               0   \n",
       "S151_nCoV004EUHM-Draw-1               0               1               3   \n",
       "\n",
       "ENSEMBLID               ENSG00000243485 ENSG00000284332 ENSG00000237613  \\\n",
       "S145_nCOV001_C                        2               0               0   \n",
       "S147_nCoV001EUHM-Draw-1               0               0               0   \n",
       "S149_nCoV002EUHM-Draw-2               0               0               0   \n",
       "S150_nCoV003EUHM-Draw-1               1               0               0   \n",
       "S151_nCoV004EUHM-Draw-1               0               0               0   \n",
       "\n",
       "ENSEMBLID               ENSG00000268020 ENSG00000240361 ENSG00000186092  \\\n",
       "S145_nCOV001_C                        0               1               0   \n",
       "S147_nCoV001EUHM-Draw-1               0               0               0   \n",
       "S149_nCoV002EUHM-Draw-2               0               0               0   \n",
       "S150_nCoV003EUHM-Draw-1               0               0               0   \n",
       "S151_nCoV004EUHM-Draw-1               0               0               0   \n",
       "\n",
       "ENSEMBLID               ENSG00000238009  ... ENSG00000273532 ENSG00000276351  \\\n",
       "S145_nCOV001_C                        1  ...               0               0   \n",
       "S147_nCoV001EUHM-Draw-1               0  ...               0               0   \n",
       "S149_nCoV002EUHM-Draw-2               0  ...               0               0   \n",
       "S150_nCoV003EUHM-Draw-1               0  ...               0               0   \n",
       "S151_nCoV004EUHM-Draw-1               0  ...               0               0   \n",
       "\n",
       "ENSEMBLID               ENSG00000275661 ENSG00000277856 ENSG00000275063  \\\n",
       "S145_nCOV001_C                        0              38              56   \n",
       "S147_nCoV001EUHM-Draw-1               0             174             151   \n",
       "S149_nCoV002EUHM-Draw-2               0             283            1313   \n",
       "S150_nCoV003EUHM-Draw-1               0            1365            5028   \n",
       "S151_nCoV004EUHM-Draw-1               0             184            2240   \n",
       "\n",
       "ENSEMBLID               ENSG00000271254 ENSG00000275405 ENSG00000275987  \\\n",
       "S145_nCOV001_C                       40              31               0   \n",
       "S147_nCoV001EUHM-Draw-1              31              22               0   \n",
       "S149_nCoV002EUHM-Draw-2              56              13               0   \n",
       "S150_nCoV003EUHM-Draw-1              42              19               0   \n",
       "S151_nCoV004EUHM-Draw-1              42               5               0   \n",
       "\n",
       "ENSEMBLID               ENSG00000277475 ENSG00000268674  \n",
       "S145_nCOV001_C                        0               0  \n",
       "S147_nCoV001EUHM-Draw-1               0               0  \n",
       "S149_nCoV002EUHM-Draw-2               0               0  \n",
       "S150_nCoV003EUHM-Draw-1               0               0  \n",
       "S151_nCoV004EUHM-Draw-1               1               0  \n",
       "\n",
       "[5 rows x 60683 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_file = gse152418_folder + '/GSE152418_p20047_Study1_RawCounts.txt'\n",
    "counts_gse152418 = pd.read_csv(counts_file, sep='\\t')\n",
    "counts_gse152418_T = counts_gse152418.T\n",
    "counts_gse152418_T.columns = counts_gse152418_T.iloc[0,:]\n",
    "counts_gse152418_T = counts_gse152418_T.drop(counts_gse152418_T.index[0])\n",
    "print(counts_gse152418.shape)\n",
    "\n",
    "counts_gse152418_T.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(34, 44)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/eliasschreiner/PROJEKT/utils/geo_utils.py:46: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  df = df.applymap(lambda x: x.strip('\"') if isinstance(x, str) else x)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>contact_zip/postal_code</th>\n",
       "      <th>contact_country</th>\n",
       "      <th>data_row_count</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>relation</th>\n",
       "      <th>relation.2</th>\n",
       "      <th>supplementary_file_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S145_nCOV001_C</td>\n",
       "      <td>GSM4614985</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 40</td>\n",
       "      <td>...</td>\n",
       "      <td>30329-4208</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 2</td>\n",
       "      <td>...</td>\n",
       "      <td>30329-4208</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 23</td>\n",
       "      <td>...</td>\n",
       "      <td>30329-4208</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 15</td>\n",
       "      <td>...</td>\n",
       "      <td>30329-4208</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 9</td>\n",
       "      <td>...</td>\n",
       "      <td>30329-4208</td>\n",
       "      <td>USA</td>\n",
       "      <td>0</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     title geo_accession                 status  \\\n",
       "0           S145_nCOV001_C    GSM4614985  Public on Jul 31 2020   \n",
       "1  S147_nCoV001EUHM-Draw-1    GSM4614986  Public on Jul 31 2020   \n",
       "2  S149_nCoV002EUHM-Draw-2    GSM4614987  Public on Jul 31 2020   \n",
       "3  S150_nCoV003EUHM-Draw-1    GSM4614988  Public on Jul 31 2020   \n",
       "4  S151_nCoV004EUHM-Draw-1    GSM4614989  Public on Jul 31 2020   \n",
       "\n",
       "  submission_date last_update_date type channel_count source_name_ch1  \\\n",
       "0     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "1     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "2     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "3     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "4     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "\n",
       "   organism_ch1          characteristics_ch1  ... contact_zip/postal_code  \\\n",
       "0  Homo sapiens  days_post_symptom_onset: 40  ...              30329-4208   \n",
       "1  Homo sapiens   days_post_symptom_onset: 2  ...              30329-4208   \n",
       "2  Homo sapiens  days_post_symptom_onset: 23  ...              30329-4208   \n",
       "3  Homo sapiens  days_post_symptom_onset: 15  ...              30329-4208   \n",
       "4  Homo sapiens   days_post_symptom_onset: 9  ...              30329-4208   \n",
       "\n",
       "  contact_country data_row_count       instrument_model library_selection  \\\n",
       "0             USA              0  Illumina NovaSeq 6000              cDNA   \n",
       "1             USA              0  Illumina NovaSeq 6000              cDNA   \n",
       "2             USA              0  Illumina NovaSeq 6000              cDNA   \n",
       "3             USA              0  Illumina NovaSeq 6000              cDNA   \n",
       "4             USA              0  Illumina NovaSeq 6000              cDNA   \n",
       "\n",
       "   library_source library_strategy  \\\n",
       "0  transcriptomic          RNA-Seq   \n",
       "1  transcriptomic          RNA-Seq   \n",
       "2  transcriptomic          RNA-Seq   \n",
       "3  transcriptomic          RNA-Seq   \n",
       "4  transcriptomic          RNA-Seq   \n",
       "\n",
       "                                            relation  \\\n",
       "0  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "1  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "2  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "3  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "4  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "\n",
       "                                          relation.2 supplementary_file_1  \n",
       "0  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "1  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "2  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "3  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "4  SRA: https://www.ncbi.nlm.nih.gov/sra?term=SRX...                 NONE  \n",
       "\n",
       "[5 rows x 44 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "metadata_txt_GSE152418 = read_prefixes_from_metadata_txt(f\"{gse152418_folder}/GSE152418_series_matrix.txt\")\n",
    "print(metadata_txt_GSE152418.shape)\n",
    "metadata_txt_GSE152418.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(34, 60727)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273532</th>\n",
       "      <th>ENSG00000276351</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S145_nCOV001_C</td>\n",
       "      <td>GSM4614985</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 40</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 23</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 9</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60727 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     title geo_accession                 status  \\\n",
       "0           S145_nCOV001_C    GSM4614985  Public on Jul 31 2020   \n",
       "1  S147_nCoV001EUHM-Draw-1    GSM4614986  Public on Jul 31 2020   \n",
       "2  S149_nCoV002EUHM-Draw-2    GSM4614987  Public on Jul 31 2020   \n",
       "3  S150_nCoV003EUHM-Draw-1    GSM4614988  Public on Jul 31 2020   \n",
       "4  S151_nCoV004EUHM-Draw-1    GSM4614989  Public on Jul 31 2020   \n",
       "\n",
       "  submission_date last_update_date type channel_count source_name_ch1  \\\n",
       "0     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "1     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "2     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "3     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "4     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "\n",
       "   organism_ch1          characteristics_ch1  ... ENSG00000273532  \\\n",
       "0  Homo sapiens  days_post_symptom_onset: 40  ...               0   \n",
       "1  Homo sapiens   days_post_symptom_onset: 2  ...               0   \n",
       "2  Homo sapiens  days_post_symptom_onset: 23  ...               0   \n",
       "3  Homo sapiens  days_post_symptom_onset: 15  ...               0   \n",
       "4  Homo sapiens   days_post_symptom_onset: 9  ...               0   \n",
       "\n",
       "  ENSG00000276351 ENSG00000275661 ENSG00000277856 ENSG00000275063  \\\n",
       "0               0               0              38              56   \n",
       "1               0               0             174             151   \n",
       "2               0               0             283            1313   \n",
       "3               0               0            1365            5028   \n",
       "4               0               0             184            2240   \n",
       "\n",
       "  ENSG00000271254 ENSG00000275405 ENSG00000275987 ENSG00000277475  \\\n",
       "0              40              31               0               0   \n",
       "1              31              22               0               0   \n",
       "2              56              13               0               0   \n",
       "3              42              19               0               0   \n",
       "4              42               5               0               1   \n",
       "\n",
       "  ENSG00000268674  \n",
       "0               0  \n",
       "1               0  \n",
       "2               0  \n",
       "3               0  \n",
       "4               0  \n",
       "\n",
       "[5 rows x 60727 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df_GSE152418 = pd.merge(metadata_txt_GSE152418, counts_gse152418_T, left_on='title', right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(34, 60727)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273532</th>\n",
       "      <th>ENSG00000276351</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S145_nCOV001_C</td>\n",
       "      <td>GSM4614985</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 40</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 23</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 9</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60727 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     title geo_accession                 status  \\\n",
       "0           S145_nCOV001_C    GSM4614985  Public on Jul 31 2020   \n",
       "1  S147_nCoV001EUHM-Draw-1    GSM4614986  Public on Jul 31 2020   \n",
       "2  S149_nCoV002EUHM-Draw-2    GSM4614987  Public on Jul 31 2020   \n",
       "3  S150_nCoV003EUHM-Draw-1    GSM4614988  Public on Jul 31 2020   \n",
       "4  S151_nCoV004EUHM-Draw-1    GSM4614989  Public on Jul 31 2020   \n",
       "\n",
       "  submission_date last_update_date type channel_count source_name_ch1  \\\n",
       "0     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "1     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "2     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "3     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "4     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "\n",
       "   organism_ch1          characteristics_ch1  ... ENSG00000273532  \\\n",
       "0  Homo sapiens  days_post_symptom_onset: 40  ...               0   \n",
       "1  Homo sapiens   days_post_symptom_onset: 2  ...               0   \n",
       "2  Homo sapiens  days_post_symptom_onset: 23  ...               0   \n",
       "3  Homo sapiens  days_post_symptom_onset: 15  ...               0   \n",
       "4  Homo sapiens   days_post_symptom_onset: 9  ...               0   \n",
       "\n",
       "  ENSG00000276351 ENSG00000275661 ENSG00000277856 ENSG00000275063  \\\n",
       "0               0               0              38              56   \n",
       "1               0               0             174             151   \n",
       "2               0               0             283            1313   \n",
       "3               0               0            1365            5028   \n",
       "4               0               0             184            2240   \n",
       "\n",
       "  ENSG00000271254 ENSG00000275405 ENSG00000275987 ENSG00000277475  \\\n",
       "0              40              31               0               0   \n",
       "1              31              22               0               0   \n",
       "2              56              13               0               0   \n",
       "3              42              19               0               0   \n",
       "4              42               5               0               1   \n",
       "\n",
       "  ENSG00000268674  \n",
       "0               0  \n",
       "1               0  \n",
       "2               0  \n",
       "3               0  \n",
       "4               0  \n",
       "\n",
       "[5 rows x 60727 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(merged_df_GSE152418.shape)\n",
    "merged_df_GSE152418.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df_GSE152418.to_csv(f\"2020_aranachalam_GSE152418_combi.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000275661</th>\n",
       "      <th>ENSG00000277856</th>\n",
       "      <th>ENSG00000275063</th>\n",
       "      <th>ENSG00000271254</th>\n",
       "      <th>ENSG00000275405</th>\n",
       "      <th>ENSG00000275987</th>\n",
       "      <th>ENSG00000277475</th>\n",
       "      <th>ENSG00000268674</th>\n",
       "      <th>ID_1</th>\n",
       "      <th>ID_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S145_nCOV001_C</td>\n",
       "      <td>GSM4614985</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 40</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>S145</td>\n",
       "      <td>nCOV001_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>S147</td>\n",
       "      <td>nCoV001EUHM-Draw-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 23</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>S149</td>\n",
       "      <td>nCoV002EUHM-Draw-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>S150</td>\n",
       "      <td>nCoV003EUHM-Draw-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 9</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>S151</td>\n",
       "      <td>nCoV004EUHM-Draw-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60729 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     title geo_accession                 status  \\\n",
       "0           S145_nCOV001_C    GSM4614985  Public on Jul 31 2020   \n",
       "1  S147_nCoV001EUHM-Draw-1    GSM4614986  Public on Jul 31 2020   \n",
       "2  S149_nCoV002EUHM-Draw-2    GSM4614987  Public on Jul 31 2020   \n",
       "3  S150_nCoV003EUHM-Draw-1    GSM4614988  Public on Jul 31 2020   \n",
       "4  S151_nCoV004EUHM-Draw-1    GSM4614989  Public on Jul 31 2020   \n",
       "\n",
       "  submission_date last_update_date type channel_count source_name_ch1  \\\n",
       "0     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "1     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "2     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "3     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "4     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "\n",
       "   organism_ch1          characteristics_ch1  ... ENSG00000275661  \\\n",
       "0  Homo sapiens  days_post_symptom_onset: 40  ...               0   \n",
       "1  Homo sapiens   days_post_symptom_onset: 2  ...               0   \n",
       "2  Homo sapiens  days_post_symptom_onset: 23  ...               0   \n",
       "3  Homo sapiens  days_post_symptom_onset: 15  ...               0   \n",
       "4  Homo sapiens   days_post_symptom_onset: 9  ...               0   \n",
       "\n",
       "  ENSG00000277856 ENSG00000275063 ENSG00000271254 ENSG00000275405  \\\n",
       "0              38              56              40              31   \n",
       "1             174             151              31              22   \n",
       "2             283            1313              56              13   \n",
       "3            1365            5028              42              19   \n",
       "4             184            2240              42               5   \n",
       "\n",
       "  ENSG00000275987 ENSG00000277475 ENSG00000268674  ID_1                ID_2  \n",
       "0               0               0               0  S145           nCOV001_C  \n",
       "1               0               0               0  S147  nCoV001EUHM-Draw-1  \n",
       "2               0               0               0  S149  nCoV002EUHM-Draw-2  \n",
       "3               0               0               0  S150  nCoV003EUHM-Draw-1  \n",
       "4               0               1               0  S151  nCoV004EUHM-Draw-1  \n",
       "\n",
       "[5 rows x 60729 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df_GSE152418[['ID_1', 'ID_2']] = merged_df_GSE152418['title'].str.split('_', expand=True, n=1)\n",
    "merged_df_GSE152418.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#### Then i need mapping for the ESEMBLEID´s to gene descriptors. I used this: https://www.biotools.fr/mouse/ensembl_symbol_converter \n",
    "ensambleid_mapping = pd.read_csv('../utils/ENSEMBLID_mapping.txt', sep='\\t')\n",
    "mapping_dict_ensemblid = ensambleid_mapping.set_index('ENSEMBLID')['Gene'].to_dict()\n",
    "counts_gse152418_T.columns = counts_gse152418_T.columns.map(mapping_dict_ensemblid)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(34, 60684)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ENSEMBLID</th>\n",
       "      <th>DDX11L1</th>\n",
       "      <th>WASH7P</th>\n",
       "      <th>MIR6859-1</th>\n",
       "      <th>MIR1302-2HG</th>\n",
       "      <th>MIR1302-2</th>\n",
       "      <th>FAM138A</th>\n",
       "      <th>OR4G4P</th>\n",
       "      <th>OR4G11P</th>\n",
       "      <th>OR4F5</th>\n",
       "      <th>AL627309.1</th>\n",
       "      <th>...</th>\n",
       "      <th>RF01973</th>\n",
       "      <th>RF01974</th>\n",
       "      <th>AC233755.2</th>\n",
       "      <th>AC233755.1</th>\n",
       "      <th>AC240274.1</th>\n",
       "      <th>RF00003</th>\n",
       "      <th>RF00003</th>\n",
       "      <th>AC213203.1</th>\n",
       "      <th>FAM231C</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM4614985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>174</td>\n",
       "      <td>151</td>\n",
       "      <td>31</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM4614986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>283</td>\n",
       "      <td>1313</td>\n",
       "      <td>56</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM4614987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1365</td>\n",
       "      <td>5028</td>\n",
       "      <td>42</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM4614988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>184</td>\n",
       "      <td>2240</td>\n",
       "      <td>42</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM4614989</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60684 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ENSEMBLID DDX11L1 WASH7P MIR6859-1 MIR1302-2HG MIR1302-2 FAM138A OR4G4P  \\\n",
       "0               0      1         3           2         0       0      0   \n",
       "1               0      0         1           0         0       0      0   \n",
       "2               0      3         2           0         0       0      0   \n",
       "3               0     16         0           1         0       0      0   \n",
       "4               0      1         3           0         0       0      0   \n",
       "\n",
       "ENSEMBLID OR4G11P OR4F5 AL627309.1  ... RF01973 RF01974 AC233755.2 AC233755.1  \\\n",
       "0               1     0          1  ...       0       0         38         56   \n",
       "1               0     0          0  ...       0       0        174        151   \n",
       "2               0     0          0  ...       0       0        283       1313   \n",
       "3               0     0          0  ...       0       0       1365       5028   \n",
       "4               0     0          0  ...       0       0        184       2240   \n",
       "\n",
       "ENSEMBLID AC240274.1 RF00003 RF00003 AC213203.1 FAM231C          ID  \n",
       "0                 40      31       0          0       0  GSM4614985  \n",
       "1                 31      22       0          0       0  GSM4614986  \n",
       "2                 56      13       0          0       0  GSM4614987  \n",
       "3                 42      19       0          0       0  GSM4614988  \n",
       "4                 42       5       0          1       0  GSM4614989  \n",
       "\n",
       "[5 rows x 60684 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(counts_gse152418_T.shape)\n",
    "counts_gse152418_T.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(211, 9)\n",
      "Unique patient ids in metadata: 158\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>Infection_status</th>\n",
       "      <th>Days_post_symptom</th>\n",
       "      <th>Day_range_post_symptom</th>\n",
       "      <th>Infection_severity</th>\n",
       "      <th>Chelation_agent</th>\n",
       "      <th>Site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>nCOV001</td>\n",
       "      <td>M</td>\n",
       "      <td>80.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>40.0</td>\n",
       "      <td>41 to 50</td>\n",
       "      <td>Convalescent</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>nCOV002</td>\n",
       "      <td>F</td>\n",
       "      <td>80.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>40.0</td>\n",
       "      <td>41 to 50</td>\n",
       "      <td>Convalescent</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>nCoV001EUHM (Draw 1)</td>\n",
       "      <td>F</td>\n",
       "      <td>75.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>2.0</td>\n",
       "      <td>01 to 10</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>nCoV002EUHM (Draw 1)</td>\n",
       "      <td>M</td>\n",
       "      <td>54.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>17.0</td>\n",
       "      <td>11 to 17</td>\n",
       "      <td>ICU</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>nCoV002EUHM (Draw 2)</td>\n",
       "      <td>M</td>\n",
       "      <td>54.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>23.0</td>\n",
       "      <td>21 to 30</td>\n",
       "      <td>Severe</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Patient_ID Gender   Age Infection_status  Days_post_symptom  \\\n",
       "73               nCOV001      M  80.0          CoVID19               40.0   \n",
       "74               nCOV002      F  80.0          CoVID19               40.0   \n",
       "75  nCoV001EUHM (Draw 1)      F  75.0          CoVID19                2.0   \n",
       "76  nCoV002EUHM (Draw 1)      M  54.0          CoVID19               17.0   \n",
       "77  nCoV002EUHM (Draw 2)      M  54.0          CoVID19               23.0   \n",
       "\n",
       "   Day_range_post_symptom Infection_severity Chelation_agent   Site  \n",
       "73               41 to 50       Convalescent         Citrate  Emory  \n",
       "74               41 to 50       Convalescent         Citrate  Emory  \n",
       "75               01 to 10           Moderate         Citrate  Emory  \n",
       "76               11 to 17                ICU         Citrate  Emory  \n",
       "77               21 to 30             Severe         Citrate  Emory  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### I got data per mail from Flo Wimmers: \n",
    "metadata_flo_gse152418_path = \"/home/eliasschreiner/PROJEKT/2020_Arunachalam_Wimmers/arunachalam_subjects.xlsx\"\n",
    "metadata_gse152418 = pd.read_excel(metadata_flo_gse152418_path)\n",
    "print(metadata_gse152418.shape)\n",
    "print(f\"Unique patient ids in metadata: {len(list(metadata_gse152418['Patient_ID'].unique()))}\")\n",
    "metadata_gse152418.iloc[74:111,:]\n",
    "\n",
    "metadata_gse152418.head()\n",
    "metadata_gse152418.iloc[73:112,:].head()\n",
    "# Conditionally transform the 'original_id' column to match the 'ID_2' format only if it contains \"(Draw X)\"\n",
    "metadata_gse152418['ID_2'] = metadata_gse152418['Patient_ID'].astype(str).apply(\n",
    "    lambda x: x if pd.isnull(x) or '(Draw ' not in x else x.replace(' (Draw ', '-Draw-').replace(')', '')\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(35, 60739)\n",
      "(24, 60739)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>characteristics_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>Infection_status</th>\n",
       "      <th>Days_post_symptom</th>\n",
       "      <th>Day_range_post_symptom</th>\n",
       "      <th>Infection_severity</th>\n",
       "      <th>Chelation_agent</th>\n",
       "      <th>Site</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S147_nCoV001EUHM-Draw-1</td>\n",
       "      <td>GSM4614986</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 2</td>\n",
       "      <td>...</td>\n",
       "      <td>nCoV001EUHM (Draw 1)</td>\n",
       "      <td>F</td>\n",
       "      <td>75.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>2.0</td>\n",
       "      <td>01 to 10</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S149_nCoV002EUHM-Draw-2</td>\n",
       "      <td>GSM4614987</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 23</td>\n",
       "      <td>...</td>\n",
       "      <td>nCoV002EUHM (Draw 2)</td>\n",
       "      <td>M</td>\n",
       "      <td>54.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>23.0</td>\n",
       "      <td>21 to 30</td>\n",
       "      <td>Severe</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S150_nCoV003EUHM-Draw-1</td>\n",
       "      <td>GSM4614988</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 15</td>\n",
       "      <td>...</td>\n",
       "      <td>nCoV003EUHM (Draw 1)</td>\n",
       "      <td>F</td>\n",
       "      <td>75.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>15.0</td>\n",
       "      <td>11 to 20</td>\n",
       "      <td>Severe</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S151_nCoV004EUHM-Draw-1</td>\n",
       "      <td>GSM4614989</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 9</td>\n",
       "      <td>...</td>\n",
       "      <td>nCoV004EUHM (Draw 1)</td>\n",
       "      <td>M</td>\n",
       "      <td>59.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>9.0</td>\n",
       "      <td>01 to 10</td>\n",
       "      <td>ICU</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>S152_nCoV006EUHM-Draw-1</td>\n",
       "      <td>GSM4614990</td>\n",
       "      <td>Public on Jul 31 2020</td>\n",
       "      <td>Jun 13 2020</td>\n",
       "      <td>Feb 01 2021</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>days_post_symptom_onset: 16</td>\n",
       "      <td>...</td>\n",
       "      <td>nCoV006EUHM (Draw 1)</td>\n",
       "      <td>M</td>\n",
       "      <td>59.0</td>\n",
       "      <td>CoVID19</td>\n",
       "      <td>16.0</td>\n",
       "      <td>11 to 20</td>\n",
       "      <td>Severe</td>\n",
       "      <td>Citrate</td>\n",
       "      <td>Emory</td>\n",
       "      <td>both</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 60739 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     title geo_accession                 status  \\\n",
       "1  S147_nCoV001EUHM-Draw-1    GSM4614986  Public on Jul 31 2020   \n",
       "2  S149_nCoV002EUHM-Draw-2    GSM4614987  Public on Jul 31 2020   \n",
       "3  S150_nCoV003EUHM-Draw-1    GSM4614988  Public on Jul 31 2020   \n",
       "4  S151_nCoV004EUHM-Draw-1    GSM4614989  Public on Jul 31 2020   \n",
       "5  S152_nCoV006EUHM-Draw-1    GSM4614990  Public on Jul 31 2020   \n",
       "\n",
       "  submission_date last_update_date type channel_count source_name_ch1  \\\n",
       "1     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "2     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "3     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "4     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "5     Jun 13 2020      Feb 01 2021  SRA             1            PBMC   \n",
       "\n",
       "   organism_ch1          characteristics_ch1  ...            Patient_ID  \\\n",
       "1  Homo sapiens   days_post_symptom_onset: 2  ...  nCoV001EUHM (Draw 1)   \n",
       "2  Homo sapiens  days_post_symptom_onset: 23  ...  nCoV002EUHM (Draw 2)   \n",
       "3  Homo sapiens  days_post_symptom_onset: 15  ...  nCoV003EUHM (Draw 1)   \n",
       "4  Homo sapiens   days_post_symptom_onset: 9  ...  nCoV004EUHM (Draw 1)   \n",
       "5  Homo sapiens  days_post_symptom_onset: 16  ...  nCoV006EUHM (Draw 1)   \n",
       "\n",
       "  Gender   Age Infection_status Days_post_symptom Day_range_post_symptom  \\\n",
       "1      F  75.0          CoVID19               2.0               01 to 10   \n",
       "2      M  54.0          CoVID19              23.0               21 to 30   \n",
       "3      F  75.0          CoVID19              15.0               11 to 20   \n",
       "4      M  59.0          CoVID19               9.0               01 to 10   \n",
       "5      M  59.0          CoVID19              16.0               11 to 20   \n",
       "\n",
       "  Infection_severity Chelation_agent   Site _merge  \n",
       "1           Moderate         Citrate  Emory   both  \n",
       "2             Severe         Citrate  Emory   both  \n",
       "3             Severe         Citrate  Emory   both  \n",
       "4                ICU         Citrate  Emory   both  \n",
       "5             Severe         Citrate  Emory   both  \n",
       "\n",
       "[5 rows x 60739 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df_GSE152418_w_age_df_left = pd.merge(merged_df_GSE152418, metadata_gse152418, on='ID_2', how='left', indicator=True)\n",
    "print(merged_df_GSE152418_w_age_df.shape)\n",
    "merged_df_GSE152418_w_age_df[merged_df_GSE152418_w_age_df['_merge'] == 'left_only']['title']\n",
    "# unmergeable: \n",
    "''' \n",
    "0               S145_nCOV001_C\n",
    "7     S154_nCoV0010EUHM-Draw-1\n",
    "8             S155_nCOV021EUHM\n",
    "9      S156_nCOV024EUHM-Draw-1\n",
    "10           S157_nCOV0029EUHM\n",
    "11     S175_nCoV024EUHM-Draw-2\n",
    "15     S179_nCoV033EUHM-Draw-1\n",
    "16     S180_nCoV034EUHM-Draw-1\n",
    "30                 S182_SHXA10\n",
    "32                 S184_SHXA18\n",
    "34                 S186_SHXA14\n",
    "'''\n",
    "### I guess i could manually add: S145_nCOV001_C, S154_nCoV0010EUHM-Draw-1 and S157_nCOV0029EUHM(probably a type and a 0 was added), S155_nCOV021EUHM (as there is only 1 draw), \n",
    "merged_df_GSE152418_w_age_df_matches = merged_df_GSE152418_w_age_df[merged_df_GSE152418_w_age_df['_merge'] != 'left_only']\n",
    "print(merged_df_GSE152418_w_age_df_matches.shape)\n",
    "merged_df_GSE152418_w_age_df_matches.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df_GSE152418_w_age_df_matches.to_csv(\"2020_aranachalam_GSE152418_combi.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(144, 33)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Age</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>...</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>Sample_name</th>\n",
       "      <th>sex</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR12404152</td>\n",
       "      <td>75</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>127</td>\n",
       "      <td>4705722618</td>\n",
       "      <td>PRJNA655740</td>\n",
       "      <td>SAMN15751363</td>\n",
       "      <td>2422148361</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>GEO</td>\n",
       "      <td>...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2020-08-09T00:00:00Z</td>\n",
       "      <td>2020-08-06T22:04:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4712885</td>\n",
       "      <td>nCOV3EUHM</td>\n",
       "      <td>female</td>\n",
       "      <td>single cell mRNA</td>\n",
       "      <td>SRP276502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR12404153</td>\n",
       "      <td>75</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>127</td>\n",
       "      <td>4774315699</td>\n",
       "      <td>PRJNA655740</td>\n",
       "      <td>SAMN15751363</td>\n",
       "      <td>2508685744</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>GEO</td>\n",
       "      <td>...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2020-08-09T00:00:00Z</td>\n",
       "      <td>2020-08-06T21:59:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4712885</td>\n",
       "      <td>nCOV3EUHM</td>\n",
       "      <td>female</td>\n",
       "      <td>single cell mRNA</td>\n",
       "      <td>SRP276502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR12404154</td>\n",
       "      <td>75</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>127</td>\n",
       "      <td>4845187922</td>\n",
       "      <td>PRJNA655740</td>\n",
       "      <td>SAMN15751363</td>\n",
       "      <td>2484581412</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>GEO</td>\n",
       "      <td>...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2020-08-09T00:00:00Z</td>\n",
       "      <td>2020-08-06T22:07:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4712885</td>\n",
       "      <td>nCOV3EUHM</td>\n",
       "      <td>female</td>\n",
       "      <td>single cell mRNA</td>\n",
       "      <td>SRP276502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR12404156</td>\n",
       "      <td>75</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>127</td>\n",
       "      <td>4939458244</td>\n",
       "      <td>PRJNA655740</td>\n",
       "      <td>SAMN15751363</td>\n",
       "      <td>2527824162</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>GEO</td>\n",
       "      <td>...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2020-08-09T00:00:00Z</td>\n",
       "      <td>2020-08-06T22:25:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4712885</td>\n",
       "      <td>nCOV3EUHM</td>\n",
       "      <td>female</td>\n",
       "      <td>single cell mRNA</td>\n",
       "      <td>SRP276502</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR12404159</td>\n",
       "      <td>60</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>127</td>\n",
       "      <td>3810265049</td>\n",
       "      <td>PRJNA655740</td>\n",
       "      <td>SAMN15751349</td>\n",
       "      <td>1967331144</td>\n",
       "      <td>PBMC</td>\n",
       "      <td>GEO</td>\n",
       "      <td>...</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2020-08-09T00:00:00Z</td>\n",
       "      <td>2020-08-06T21:26:00Z</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM4712899</td>\n",
       "      <td>nCOV021EUHM</td>\n",
       "      <td>female</td>\n",
       "      <td>single cell mRNA</td>\n",
       "      <td>SRP276502</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Run  Age Assay Type  AvgSpotLen       Bases   BioProject  \\\n",
       "0  SRR12404152   75    RNA-Seq         127  4705722618  PRJNA655740   \n",
       "1  SRR12404153   75    RNA-Seq         127  4774315699  PRJNA655740   \n",
       "2  SRR12404154   75    RNA-Seq         127  4845187922  PRJNA655740   \n",
       "3  SRR12404156   75    RNA-Seq         127  4939458244  PRJNA655740   \n",
       "4  SRR12404159   60    RNA-Seq         127  3810265049  PRJNA655740   \n",
       "\n",
       "      BioSample       Bytes cell_type Center Name  ...      Organism  \\\n",
       "0  SAMN15751363  2422148361      PBMC         GEO  ...  Homo sapiens   \n",
       "1  SAMN15751363  2508685744      PBMC         GEO  ...  Homo sapiens   \n",
       "2  SAMN15751363  2484581412      PBMC         GEO  ...  Homo sapiens   \n",
       "3  SAMN15751363  2527824162      PBMC         GEO  ...  Homo sapiens   \n",
       "4  SAMN15751349  1967331144      PBMC         GEO  ...  Homo sapiens   \n",
       "\n",
       "   Platform           ReleaseDate           create_date  version Sample Name  \\\n",
       "0  ILLUMINA  2020-08-09T00:00:00Z  2020-08-06T22:04:00Z        1  GSM4712885   \n",
       "1  ILLUMINA  2020-08-09T00:00:00Z  2020-08-06T21:59:00Z        1  GSM4712885   \n",
       "2  ILLUMINA  2020-08-09T00:00:00Z  2020-08-06T22:07:00Z        1  GSM4712885   \n",
       "3  ILLUMINA  2020-08-09T00:00:00Z  2020-08-06T22:25:00Z        1  GSM4712885   \n",
       "4  ILLUMINA  2020-08-09T00:00:00Z  2020-08-06T21:26:00Z        1  GSM4712899   \n",
       "\n",
       "   Sample_name     sex       source_name  SRA Study  \n",
       "0    nCOV3EUHM  female  single cell mRNA  SRP276502  \n",
       "1    nCOV3EUHM  female  single cell mRNA  SRP276502  \n",
       "2    nCOV3EUHM  female  single cell mRNA  SRP276502  \n",
       "3    nCOV3EUHM  female  single cell mRNA  SRP276502  \n",
       "4  nCOV021EUHM  female  single cell mRNA  SRP276502  \n",
       "\n",
       "[5 rows x 33 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_gse152418\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# ### Metadata from the SRA Run tool -> i feel like this metadata is wrong... . Currently the SRA selector tool lacka age... \n",
    "# -> Flo fragen ob er hier noch infos zu der subjects ages hat \n",
    "\n",
    "\n",
    "## was for the other GSE sample which is single cell. .\n",
    "# metadata_file = gse152418_folder + '/metadata.txt'\n",
    "# metadata_gse152418 = pd.read_csv(metadata_file)\n",
    "# print(metadata_gse152418.shape)\n",
    "# metadata_gse152418.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Age</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>...</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>Sample_name</th>\n",
       "      <th>sex</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Run, Age, Assay Type, AvgSpotLen, Bases, BioProject, BioSample, Bytes, cell_type, Center Name, Consent, DATASTORE filetype, DATASTORE provider, DATASTORE region, days_since_symptom_onset, disease_severity, disease_status, Experiment, GEO_Accession (exp), Instrument, LibraryLayout, LibrarySelection, LibrarySource, Organism, Platform, ReleaseDate, create_date, version, Sample Name, Sample_name, sex, source_name, SRA Study]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 33 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Oooookay my mapping did not work out, the GSM from meta differs from the downloaded table\n",
    "metadata_gse152418[metadata_gse152418['Sample Name']=='GSM4614987']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, 60717)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DDX11L1</th>\n",
       "      <th>WASH7P</th>\n",
       "      <th>MIR6859-1</th>\n",
       "      <th>MIR1302-2HG</th>\n",
       "      <th>MIR1302-2</th>\n",
       "      <th>FAM138A</th>\n",
       "      <th>OR4G4P</th>\n",
       "      <th>OR4G11P</th>\n",
       "      <th>OR4F5</th>\n",
       "      <th>AL627309.1</th>\n",
       "      <th>...</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>create_date</th>\n",
       "      <th>version</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>Sample_name</th>\n",
       "      <th>sex</th>\n",
       "      <th>source_name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 60717 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [DDX11L1, WASH7P, MIR6859-1, MIR1302-2HG, MIR1302-2, FAM138A, OR4G4P, OR4G11P, OR4F5, AL627309.1, AL627309.3, CICP27, AL627309.6, AL627309.7, AL627309.2, AL627309.5, RNU6-1100P, AL627309.4, FO538757.2, FO538757.1, MIR6859-2, AP006222.1, nan, RPL23AP24, AL732372.1, AL732372.2, WBP1LP7, OR4F29, CICP7, AL732372.3, RF00026, AL669831.3, AC114498.1, MTND1P23, MTND2P28, MTCO1P12, MIR6723, MTCO2P12, MTATP8P1, MTATP6P1, MTCO3P12, WBP1LP6, OR4F16, CICP3, AL669831.1, RNU6-1199P, AL669831.2, AL669831.5, AL669831.7, AL669831.4, FAM87B, LINC01128, LINC00115, nan, FAM41C, TUBB8P11, AL669831.6, AL645608.7, AL645608.3, AL645608.5, AL645608.1, SAMD11, NOC2L, KLHL17, PLEKHN1, PERM1, AL645608.8, HES4, ISG15, AL645608.4, AL645608.2, AGRN, AL645608.6, AL390719.1, AL645608.9, RNF223, C1orf159, nan, LINC01342, MIR200B, MIR200A, MIR429, AL390719.2, TTLL10-AS1, TTLL10, TNFRSF18, TNFRSF4, SDF4, B3GALT6, C1QTNF12, AL162741.1, UBE2J2, LINC01786, SCNN1D, ACAP3, MIR6726, PUSL1, INTS11, MIR6727, AL139287.1, ...]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 60717 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### IMPORTANT; We merge per sample, NOT per subject here! \n",
    "combi_df_all_meta = pd.merge(counts_gse152418_T, metadata_gse152418, left_on='ID', right_on='Sample Name', how='left')\n",
    "print(combi_df_all_meta.shape)\n",
    "combi_df_all_meta.head()\n",
    "#### großen all meta file saven\n",
    "#### reduzieren um mit 10kimmu zu matchen\n",
    "\n",
    "\n",
    "#### dann evtl \"optimieren\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/var/datasets/single_papers/2020_Arunachalam_Wimmers/GSE152418/geo/.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[11], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m totalseq_df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgse152418_folder\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mgeo/.csv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    936\u001b[0m     dialect,\n\u001b[1;32m    937\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    944\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m    945\u001b[0m )\n\u001b[1;32m    946\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 948\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:611\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    608\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    610\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 611\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    613\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    614\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1448\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1445\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1447\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1448\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/pandas/io/parsers/readers.py:1705\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1703\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1704\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1705\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1706\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1707\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1708\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1709\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1710\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1711\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1712\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1713\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1714\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1715\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1716\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/micromamba/envs/merge_env/lib/python3.12/site-packages/pandas/io/common.py:863\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    858\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    859\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    860\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    861\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    862\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 863\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    864\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    865\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    866\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    867\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    868\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    869\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    870\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    871\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    872\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/var/datasets/single_papers/2020_Arunachalam_Wimmers/GSE152418/geo/.csv'"
     ]
    }
   ],
   "source": [
    "totalseq_df = pd.read_csv(gse152418_folder + 'geo/.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Biomarkers: 38\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array(['CD279 (PD-1)', 'CD86', 'CD274 (B7-H1, PD-L1)', 'CD3',\n",
       "       'CD56 (NCAM)', 'CD19', 'CD33', 'CD11c', 'CD34', 'CD45RA', 'CD123',\n",
       "       'CD8a', 'CD14', 'CD16', 'CD25', 'CD45RO',\n",
       "       'Mouse IgG1, κ Isotype Control', 'Mouse IgG2a, κ Isotype Control',\n",
       "       'Mouse IgG2b, κ Isotype Control', 'Rat IgG2b, κ Isotype Control',\n",
       "       'TCR γ/δ', 'CD103', 'CD69', 'CD197 (CCR7)', 'CD27', 'CD95 (Fas)',\n",
       "       'HLA-DR', 'CD1c', 'CD370 (CLEC9A/DNGR1)', 'FceRIa', 'CD163',\n",
       "       'CD28', 'CD38', 'CD127 (IL-7Ra)', 'TCR Va7.2', 'CD94',\n",
       "       'phycoerythrin (PE)', 'CD4'], dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Biomarkers from \n",
    "print(f'Biomarkers: {len(totalseq_df['Specificity'].unique())}')\n",
    "spec_names = totalseq_df['Specificity'].unique()\n",
    "totalseq_df['Specificity'].unique()\n",
    "# what is: \n",
    "## Catalog (table headline), and Clone ?\n",
    "## κ Isotype Control ?\n",
    "# Not important but: Can you even count an antibody as a biomarker ? \n",
    "# Count try: {entry: totalseq_df['Specificity'].value_counts().get(entry, 0) for entry in spec_names}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(33576, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG00000243485</th>\n",
       "      <th>MIR1302-2HG</th>\n",
       "      <th>Gene Expression</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000239945</td>\n",
       "      <td>AL627309.3</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>AL627309.2</td>\n",
       "      <td>Gene Expression</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ENSG00000243485 MIR1302-2HG  Gene Expression\n",
       "0  ENSG00000237613     FAM138A  Gene Expression\n",
       "1  ENSG00000186092       OR4F5  Gene Expression\n",
       "2  ENSG00000238009  AL627309.1  Gene Expression\n",
       "3  ENSG00000239945  AL627309.3  Gene Expression\n",
       "4  ENSG00000239906  AL627309.2  Gene Expression"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features_df = pd.read_csv(gse152418_folder + 'geo/GSE155673_features.tsv', sep='\\t')\n",
    "print(features_df.shape)\n",
    "features_df.head()\n",
    "#### HAS NO COLUMNSNAMES -> 1. Gene ID Ensample id, 2. Gene symbol , 3. Descriptor ?\n",
    "### scTranscriptsome pro proble gibts pro zelle einen wert.\n",
    "###### spalte: sc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAACCCACACCAGCGT-1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAACCCACATCTCAAG-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAACGAAAGACCTGGA-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAACGCTCAGTGGGTA-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAACGCTGTAGCTTGT-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAACGCTGTTTGACAC-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AAACCCACACCAGCGT-1\n",
       "0  AAACCCACATCTCAAG-1\n",
       "1  AAACGAAAGACCTGGA-1\n",
       "2  AAACGCTCAGTGGGTA-1\n",
       "3  AAACGCTGTAGCTTGT-1\n",
       "4  AAACGCTGTTTGACAC-1"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "barcodes_df = pd.read_csv(gse152418_folder + 'geo/GSE155673_barcodes.tsv', sep='\\t')\n",
    "barcodes_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AAACCCACACCAGCGT-1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAACCCACATCTCAAG-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAACGAAAGACCTGGA-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAACGCTCAGTGGGTA-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAACGCTGTAGCTTGT-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAACGCTGTTTGACAC-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AAACCCACACCAGCGT-1\n",
       "0  AAACCCACATCTCAAG-1\n",
       "1  AAACGAAAGACCTGGA-1\n",
       "2  AAACGCTCAGTGGGTA-1\n",
       "3  AAACGCTGTAGCTTGT-1\n",
       "4  AAACGCTGTTTGACAC-1"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### COV 1 -> this is a subject\n",
    "cov1_barcodes_df =  pd.read_csv(gse152418_folder + 'geo/GSE155673_cov01_barcodes.tsv', sep='\\t')\n",
    "cov1_barcodes_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(33577, 4491)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>4481</th>\n",
       "      <th>4482</th>\n",
       "      <th>4483</th>\n",
       "      <th>4484</th>\n",
       "      <th>4485</th>\n",
       "      <th>4486</th>\n",
       "      <th>4487</th>\n",
       "      <th>4488</th>\n",
       "      <th>4489</th>\n",
       "      <th>4490</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 4491 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   0     1     2     3     4     5     6     7     8     9     ...  4481  \\\n",
       "0     0     0     0     0     0     0     0     0     0     0  ...     0   \n",
       "1     0     0     0     0     0     0     0     0     0     0  ...     0   \n",
       "2     0     0     0     0     0     0     0     0     0     0  ...     0   \n",
       "3     0     0     0     0     0     0     0     0     0     0  ...     0   \n",
       "4     0     0     0     0     0     0     0     0     0     0  ...     0   \n",
       "\n",
       "   4482  4483  4484  4485  4486  4487  4488  4489  4490  \n",
       "0     0     0     0     0     0     0     0     0     0  \n",
       "1     0     0     0     0     0     0     0     0     0  \n",
       "2     0     0     0     0     0     0     0     0     0  \n",
       "3     0     0     0     0     0     0     0     0     0  \n",
       "4     0     0     0     0     0     0     0     0     0  \n",
       "\n",
       "[5 rows x 4491 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sparse_matrix = mmread(gse152418_folder+ 'geo/GSE155673_cov01_matrix.mtx')\n",
    "dense_matrix = sparse_matrix.todense()\n",
    "cov1_matrix_as_df = pd.DataFrame(dense_matrix)\n",
    "print(cov1_matrix_as_df.shape)\n",
    "cov1_matrix_as_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
